{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 332,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/admin/Documents/support/dros_200MB_real_bootstrap\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "#os.chdir('..')\n",
    "dirpath = os.getcwd()\n",
    "#directory_path = \"/Users/admin/Documents/Skimming/tree_of_life\"\n",
    "#report_directory = \"report_kraken_viral_dist_unmasked_0.5_corrected_k35\"\n",
    "#os.chdir(\"/Users/admin/Documents/local_sens_hash/cpp_results/test_sw/15_anopheles_prj\")\n",
    "print (os.getcwd())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import DataFrame\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Resample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "metadata": {},
   "outputs": [],
   "source": [
    "#get filelist\n",
    "\n",
    "#report_directory = \"d0.10\"\n",
    "#report_directory = \"d0.15\"\n",
    "report_directory = \"di_mtrx\"\n",
    "\n",
    "\n",
    "#os.chdir(report_directory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "file_lst = glob.glob(report_directory+\"/di*.txt\")\n",
    "file_lst.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "101\n"
     ]
    }
   ],
   "source": [
    "print(len(file_lst))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['di_mtrx/dimtrx_main.txt']"
      ]
     },
     "execution_count": 337,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_lst[:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(file_lst[0], header = 0, sep='\\t', skiprows = 0, index_col=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425989_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425990_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425991_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425992_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425993_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425995_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425997_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425998_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425999_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6426000_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6426001_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6426002_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6426003_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6426004_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.118984</td>\n",
       "      <td>0.067789</td>\n",
       "      <td>0.114476</td>\n",
       "      <td>0.128712</td>\n",
       "      <td>0.121284</td>\n",
       "      <td>0.192597</td>\n",
       "      <td>0.133703</td>\n",
       "      <td>0.130147</td>\n",
       "      <td>0.190308</td>\n",
       "      <td>0.157289</td>\n",
       "      <td>0.131612</td>\n",
       "      <td>0.182641</td>\n",
       "      <td>0.135230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425990_2</td>\n",
       "      <td>0.118984</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.133006</td>\n",
       "      <td>0.105495</td>\n",
       "      <td>0.065635</td>\n",
       "      <td>0.098415</td>\n",
       "      <td>0.192280</td>\n",
       "      <td>0.154649</td>\n",
       "      <td>0.065078</td>\n",
       "      <td>0.185571</td>\n",
       "      <td>0.152304</td>\n",
       "      <td>0.068205</td>\n",
       "      <td>0.175759</td>\n",
       "      <td>0.063171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425991_2</td>\n",
       "      <td>0.067789</td>\n",
       "      <td>0.133006</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.112692</td>\n",
       "      <td>0.130679</td>\n",
       "      <td>0.127919</td>\n",
       "      <td>0.201744</td>\n",
       "      <td>0.153193</td>\n",
       "      <td>0.130754</td>\n",
       "      <td>0.194345</td>\n",
       "      <td>0.160089</td>\n",
       "      <td>0.134019</td>\n",
       "      <td>0.186412</td>\n",
       "      <td>0.136197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425992_2</td>\n",
       "      <td>0.114476</td>\n",
       "      <td>0.105495</td>\n",
       "      <td>0.112692</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.104296</td>\n",
       "      <td>0.098442</td>\n",
       "      <td>0.185979</td>\n",
       "      <td>0.151898</td>\n",
       "      <td>0.104607</td>\n",
       "      <td>0.187617</td>\n",
       "      <td>0.156525</td>\n",
       "      <td>0.106157</td>\n",
       "      <td>0.180429</td>\n",
       "      <td>0.108032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425993_2</td>\n",
       "      <td>0.128712</td>\n",
       "      <td>0.065635</td>\n",
       "      <td>0.130679</td>\n",
       "      <td>0.104296</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.097492</td>\n",
       "      <td>0.188330</td>\n",
       "      <td>0.150890</td>\n",
       "      <td>0.017648</td>\n",
       "      <td>0.181538</td>\n",
       "      <td>0.144520</td>\n",
       "      <td>0.021502</td>\n",
       "      <td>0.174222</td>\n",
       "      <td>0.065990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425995_2</td>\n",
       "      <td>0.121284</td>\n",
       "      <td>0.098415</td>\n",
       "      <td>0.127919</td>\n",
       "      <td>0.098442</td>\n",
       "      <td>0.097492</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.188214</td>\n",
       "      <td>0.148225</td>\n",
       "      <td>0.097711</td>\n",
       "      <td>0.187657</td>\n",
       "      <td>0.148615</td>\n",
       "      <td>0.099079</td>\n",
       "      <td>0.175503</td>\n",
       "      <td>0.101917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425997_2</td>\n",
       "      <td>0.192597</td>\n",
       "      <td>0.192280</td>\n",
       "      <td>0.201744</td>\n",
       "      <td>0.185979</td>\n",
       "      <td>0.188330</td>\n",
       "      <td>0.188214</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.140543</td>\n",
       "      <td>0.189119</td>\n",
       "      <td>0.133573</td>\n",
       "      <td>0.147395</td>\n",
       "      <td>0.182973</td>\n",
       "      <td>0.188911</td>\n",
       "      <td>0.191627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425998_2</td>\n",
       "      <td>0.133703</td>\n",
       "      <td>0.154649</td>\n",
       "      <td>0.153193</td>\n",
       "      <td>0.151898</td>\n",
       "      <td>0.150890</td>\n",
       "      <td>0.148225</td>\n",
       "      <td>0.140543</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.151709</td>\n",
       "      <td>0.173495</td>\n",
       "      <td>0.015556</td>\n",
       "      <td>0.150087</td>\n",
       "      <td>0.176831</td>\n",
       "      <td>0.156294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425999_2</td>\n",
       "      <td>0.130147</td>\n",
       "      <td>0.065078</td>\n",
       "      <td>0.130754</td>\n",
       "      <td>0.104607</td>\n",
       "      <td>0.017648</td>\n",
       "      <td>0.097711</td>\n",
       "      <td>0.189119</td>\n",
       "      <td>0.151709</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.189365</td>\n",
       "      <td>0.142723</td>\n",
       "      <td>0.019900</td>\n",
       "      <td>0.173301</td>\n",
       "      <td>0.065265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426000_2</td>\n",
       "      <td>0.190308</td>\n",
       "      <td>0.185571</td>\n",
       "      <td>0.194345</td>\n",
       "      <td>0.187617</td>\n",
       "      <td>0.181538</td>\n",
       "      <td>0.187657</td>\n",
       "      <td>0.133573</td>\n",
       "      <td>0.173495</td>\n",
       "      <td>0.189365</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.178485</td>\n",
       "      <td>0.150801</td>\n",
       "      <td>0.156946</td>\n",
       "      <td>0.186393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426001_2</td>\n",
       "      <td>0.157289</td>\n",
       "      <td>0.152304</td>\n",
       "      <td>0.160089</td>\n",
       "      <td>0.156525</td>\n",
       "      <td>0.144520</td>\n",
       "      <td>0.148615</td>\n",
       "      <td>0.147395</td>\n",
       "      <td>0.015556</td>\n",
       "      <td>0.142723</td>\n",
       "      <td>0.178485</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.136241</td>\n",
       "      <td>0.181266</td>\n",
       "      <td>0.153240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426002_2</td>\n",
       "      <td>0.131612</td>\n",
       "      <td>0.068205</td>\n",
       "      <td>0.134019</td>\n",
       "      <td>0.106157</td>\n",
       "      <td>0.021502</td>\n",
       "      <td>0.099079</td>\n",
       "      <td>0.182973</td>\n",
       "      <td>0.150087</td>\n",
       "      <td>0.019900</td>\n",
       "      <td>0.150801</td>\n",
       "      <td>0.136241</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.168563</td>\n",
       "      <td>0.068335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426003_2</td>\n",
       "      <td>0.182641</td>\n",
       "      <td>0.175759</td>\n",
       "      <td>0.186412</td>\n",
       "      <td>0.180429</td>\n",
       "      <td>0.174222</td>\n",
       "      <td>0.175503</td>\n",
       "      <td>0.188911</td>\n",
       "      <td>0.176831</td>\n",
       "      <td>0.173301</td>\n",
       "      <td>0.156946</td>\n",
       "      <td>0.181266</td>\n",
       "      <td>0.168563</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.129711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.135230</td>\n",
       "      <td>0.063171</td>\n",
       "      <td>0.136197</td>\n",
       "      <td>0.108032</td>\n",
       "      <td>0.065990</td>\n",
       "      <td>0.101917</td>\n",
       "      <td>0.191627</td>\n",
       "      <td>0.156294</td>\n",
       "      <td>0.065265</td>\n",
       "      <td>0.186393</td>\n",
       "      <td>0.153240</td>\n",
       "      <td>0.068335</td>\n",
       "      <td>0.129711</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             sample  ucseq_ucseq_merged_SRR6425989_2  \\\n",
       "0   ucseq_ucseq_merged_SRR6425989_2                         0.000000   \n",
       "1   ucseq_ucseq_merged_SRR6425990_2                         0.118984   \n",
       "2   ucseq_ucseq_merged_SRR6425991_2                         0.067789   \n",
       "3   ucseq_ucseq_merged_SRR6425992_2                         0.114476   \n",
       "4   ucseq_ucseq_merged_SRR6425993_2                         0.128712   \n",
       "5   ucseq_ucseq_merged_SRR6425995_2                         0.121284   \n",
       "6   ucseq_ucseq_merged_SRR6425997_2                         0.192597   \n",
       "7   ucseq_ucseq_merged_SRR6425998_2                         0.133703   \n",
       "8   ucseq_ucseq_merged_SRR6425999_2                         0.130147   \n",
       "9   ucseq_ucseq_merged_SRR6426000_2                         0.190308   \n",
       "10  ucseq_ucseq_merged_SRR6426001_2                         0.157289   \n",
       "11  ucseq_ucseq_merged_SRR6426002_2                         0.131612   \n",
       "12  ucseq_ucseq_merged_SRR6426003_2                         0.182641   \n",
       "13  ucseq_ucseq_merged_SRR6426004_2                         0.135230   \n",
       "\n",
       "    ucseq_ucseq_merged_SRR6425990_2  ucseq_ucseq_merged_SRR6425991_2  \\\n",
       "0                          0.118984                         0.067789   \n",
       "1                          0.000000                         0.133006   \n",
       "2                          0.133006                         0.000000   \n",
       "3                          0.105495                         0.112692   \n",
       "4                          0.065635                         0.130679   \n",
       "5                          0.098415                         0.127919   \n",
       "6                          0.192280                         0.201744   \n",
       "7                          0.154649                         0.153193   \n",
       "8                          0.065078                         0.130754   \n",
       "9                          0.185571                         0.194345   \n",
       "10                         0.152304                         0.160089   \n",
       "11                         0.068205                         0.134019   \n",
       "12                         0.175759                         0.186412   \n",
       "13                         0.063171                         0.136197   \n",
       "\n",
       "    ucseq_ucseq_merged_SRR6425992_2  ucseq_ucseq_merged_SRR6425993_2  \\\n",
       "0                          0.114476                         0.128712   \n",
       "1                          0.105495                         0.065635   \n",
       "2                          0.112692                         0.130679   \n",
       "3                          0.000000                         0.104296   \n",
       "4                          0.104296                         0.000000   \n",
       "5                          0.098442                         0.097492   \n",
       "6                          0.185979                         0.188330   \n",
       "7                          0.151898                         0.150890   \n",
       "8                          0.104607                         0.017648   \n",
       "9                          0.187617                         0.181538   \n",
       "10                         0.156525                         0.144520   \n",
       "11                         0.106157                         0.021502   \n",
       "12                         0.180429                         0.174222   \n",
       "13                         0.108032                         0.065990   \n",
       "\n",
       "    ucseq_ucseq_merged_SRR6425995_2  ucseq_ucseq_merged_SRR6425997_2  \\\n",
       "0                          0.121284                         0.192597   \n",
       "1                          0.098415                         0.192280   \n",
       "2                          0.127919                         0.201744   \n",
       "3                          0.098442                         0.185979   \n",
       "4                          0.097492                         0.188330   \n",
       "5                          0.000000                         0.188214   \n",
       "6                          0.188214                         0.000000   \n",
       "7                          0.148225                         0.140543   \n",
       "8                          0.097711                         0.189119   \n",
       "9                          0.187657                         0.133573   \n",
       "10                         0.148615                         0.147395   \n",
       "11                         0.099079                         0.182973   \n",
       "12                         0.175503                         0.188911   \n",
       "13                         0.101917                         0.191627   \n",
       "\n",
       "    ucseq_ucseq_merged_SRR6425998_2  ucseq_ucseq_merged_SRR6425999_2  \\\n",
       "0                          0.133703                         0.130147   \n",
       "1                          0.154649                         0.065078   \n",
       "2                          0.153193                         0.130754   \n",
       "3                          0.151898                         0.104607   \n",
       "4                          0.150890                         0.017648   \n",
       "5                          0.148225                         0.097711   \n",
       "6                          0.140543                         0.189119   \n",
       "7                          0.000000                         0.151709   \n",
       "8                          0.151709                         0.000000   \n",
       "9                          0.173495                         0.189365   \n",
       "10                         0.015556                         0.142723   \n",
       "11                         0.150087                         0.019900   \n",
       "12                         0.176831                         0.173301   \n",
       "13                         0.156294                         0.065265   \n",
       "\n",
       "    ucseq_ucseq_merged_SRR6426000_2  ucseq_ucseq_merged_SRR6426001_2  \\\n",
       "0                          0.190308                         0.157289   \n",
       "1                          0.185571                         0.152304   \n",
       "2                          0.194345                         0.160089   \n",
       "3                          0.187617                         0.156525   \n",
       "4                          0.181538                         0.144520   \n",
       "5                          0.187657                         0.148615   \n",
       "6                          0.133573                         0.147395   \n",
       "7                          0.173495                         0.015556   \n",
       "8                          0.189365                         0.142723   \n",
       "9                          0.000000                         0.178485   \n",
       "10                         0.178485                         0.000000   \n",
       "11                         0.150801                         0.136241   \n",
       "12                         0.156946                         0.181266   \n",
       "13                         0.186393                         0.153240   \n",
       "\n",
       "    ucseq_ucseq_merged_SRR6426002_2  ucseq_ucseq_merged_SRR6426003_2  \\\n",
       "0                          0.131612                         0.182641   \n",
       "1                          0.068205                         0.175759   \n",
       "2                          0.134019                         0.186412   \n",
       "3                          0.106157                         0.180429   \n",
       "4                          0.021502                         0.174222   \n",
       "5                          0.099079                         0.175503   \n",
       "6                          0.182973                         0.188911   \n",
       "7                          0.150087                         0.176831   \n",
       "8                          0.019900                         0.173301   \n",
       "9                          0.150801                         0.156946   \n",
       "10                         0.136241                         0.181266   \n",
       "11                         0.000000                         0.168563   \n",
       "12                         0.168563                         0.000000   \n",
       "13                         0.068335                         0.129711   \n",
       "\n",
       "    ucseq_ucseq_merged_SRR6426004_2  \n",
       "0                          0.135230  \n",
       "1                          0.063171  \n",
       "2                          0.136197  \n",
       "3                          0.108032  \n",
       "4                          0.065990  \n",
       "5                          0.101917  \n",
       "6                          0.191627  \n",
       "7                          0.156294  \n",
       "8                          0.065265  \n",
       "9                          0.186393  \n",
       "10                         0.153240  \n",
       "11                         0.068335  \n",
       "12                         0.129711  \n",
       "13                         0.000000  "
      ]
     },
     "execution_count": 339,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(n=15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425989_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425990_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425991_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425992_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425993_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425995_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425997_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425998_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6425999_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6426000_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6426001_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6426002_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6426003_2</th>\n",
       "      <th>ucseq_ucseq_merged_SRR6426004_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.118984</td>\n",
       "      <td>0.067789</td>\n",
       "      <td>0.114476</td>\n",
       "      <td>0.128712</td>\n",
       "      <td>0.121284</td>\n",
       "      <td>0.192597</td>\n",
       "      <td>0.133703</td>\n",
       "      <td>0.130147</td>\n",
       "      <td>0.190308</td>\n",
       "      <td>0.157289</td>\n",
       "      <td>0.131612</td>\n",
       "      <td>0.182641</td>\n",
       "      <td>0.135230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425990_2</td>\n",
       "      <td>0.118984</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.133006</td>\n",
       "      <td>0.105495</td>\n",
       "      <td>0.065635</td>\n",
       "      <td>0.098415</td>\n",
       "      <td>0.192280</td>\n",
       "      <td>0.154649</td>\n",
       "      <td>0.065078</td>\n",
       "      <td>0.185571</td>\n",
       "      <td>0.152304</td>\n",
       "      <td>0.068205</td>\n",
       "      <td>0.175759</td>\n",
       "      <td>0.063171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425991_2</td>\n",
       "      <td>0.067789</td>\n",
       "      <td>0.133006</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.112692</td>\n",
       "      <td>0.130679</td>\n",
       "      <td>0.127919</td>\n",
       "      <td>0.201744</td>\n",
       "      <td>0.153193</td>\n",
       "      <td>0.130754</td>\n",
       "      <td>0.194345</td>\n",
       "      <td>0.160089</td>\n",
       "      <td>0.134019</td>\n",
       "      <td>0.186412</td>\n",
       "      <td>0.136197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425992_2</td>\n",
       "      <td>0.114476</td>\n",
       "      <td>0.105495</td>\n",
       "      <td>0.112692</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.104296</td>\n",
       "      <td>0.098442</td>\n",
       "      <td>0.185979</td>\n",
       "      <td>0.151898</td>\n",
       "      <td>0.104607</td>\n",
       "      <td>0.187617</td>\n",
       "      <td>0.156525</td>\n",
       "      <td>0.106157</td>\n",
       "      <td>0.180429</td>\n",
       "      <td>0.108032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425993_2</td>\n",
       "      <td>0.128712</td>\n",
       "      <td>0.065635</td>\n",
       "      <td>0.130679</td>\n",
       "      <td>0.104296</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.097492</td>\n",
       "      <td>0.188330</td>\n",
       "      <td>0.150890</td>\n",
       "      <td>0.017648</td>\n",
       "      <td>0.181538</td>\n",
       "      <td>0.144520</td>\n",
       "      <td>0.021502</td>\n",
       "      <td>0.174222</td>\n",
       "      <td>0.065990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425995_2</td>\n",
       "      <td>0.121284</td>\n",
       "      <td>0.098415</td>\n",
       "      <td>0.127919</td>\n",
       "      <td>0.098442</td>\n",
       "      <td>0.097492</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.188214</td>\n",
       "      <td>0.148225</td>\n",
       "      <td>0.097711</td>\n",
       "      <td>0.187657</td>\n",
       "      <td>0.148615</td>\n",
       "      <td>0.099079</td>\n",
       "      <td>0.175503</td>\n",
       "      <td>0.101917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425997_2</td>\n",
       "      <td>0.192597</td>\n",
       "      <td>0.192280</td>\n",
       "      <td>0.201744</td>\n",
       "      <td>0.185979</td>\n",
       "      <td>0.188330</td>\n",
       "      <td>0.188214</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.140543</td>\n",
       "      <td>0.189119</td>\n",
       "      <td>0.133573</td>\n",
       "      <td>0.147395</td>\n",
       "      <td>0.182973</td>\n",
       "      <td>0.188911</td>\n",
       "      <td>0.191627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425998_2</td>\n",
       "      <td>0.133703</td>\n",
       "      <td>0.154649</td>\n",
       "      <td>0.153193</td>\n",
       "      <td>0.151898</td>\n",
       "      <td>0.150890</td>\n",
       "      <td>0.148225</td>\n",
       "      <td>0.140543</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.151709</td>\n",
       "      <td>0.173495</td>\n",
       "      <td>0.015556</td>\n",
       "      <td>0.150087</td>\n",
       "      <td>0.176831</td>\n",
       "      <td>0.156294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425999_2</td>\n",
       "      <td>0.130147</td>\n",
       "      <td>0.065078</td>\n",
       "      <td>0.130754</td>\n",
       "      <td>0.104607</td>\n",
       "      <td>0.017648</td>\n",
       "      <td>0.097711</td>\n",
       "      <td>0.189119</td>\n",
       "      <td>0.151709</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.189365</td>\n",
       "      <td>0.142723</td>\n",
       "      <td>0.019900</td>\n",
       "      <td>0.173301</td>\n",
       "      <td>0.065265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426000_2</td>\n",
       "      <td>0.190308</td>\n",
       "      <td>0.185571</td>\n",
       "      <td>0.194345</td>\n",
       "      <td>0.187617</td>\n",
       "      <td>0.181538</td>\n",
       "      <td>0.187657</td>\n",
       "      <td>0.133573</td>\n",
       "      <td>0.173495</td>\n",
       "      <td>0.189365</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.178485</td>\n",
       "      <td>0.150801</td>\n",
       "      <td>0.156946</td>\n",
       "      <td>0.186393</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            sample  ucseq_ucseq_merged_SRR6425989_2  \\\n",
       "0  ucseq_ucseq_merged_SRR6425989_2                         0.000000   \n",
       "1  ucseq_ucseq_merged_SRR6425990_2                         0.118984   \n",
       "2  ucseq_ucseq_merged_SRR6425991_2                         0.067789   \n",
       "3  ucseq_ucseq_merged_SRR6425992_2                         0.114476   \n",
       "4  ucseq_ucseq_merged_SRR6425993_2                         0.128712   \n",
       "5  ucseq_ucseq_merged_SRR6425995_2                         0.121284   \n",
       "6  ucseq_ucseq_merged_SRR6425997_2                         0.192597   \n",
       "7  ucseq_ucseq_merged_SRR6425998_2                         0.133703   \n",
       "8  ucseq_ucseq_merged_SRR6425999_2                         0.130147   \n",
       "9  ucseq_ucseq_merged_SRR6426000_2                         0.190308   \n",
       "\n",
       "   ucseq_ucseq_merged_SRR6425990_2  ucseq_ucseq_merged_SRR6425991_2  \\\n",
       "0                         0.118984                         0.067789   \n",
       "1                         0.000000                         0.133006   \n",
       "2                         0.133006                         0.000000   \n",
       "3                         0.105495                         0.112692   \n",
       "4                         0.065635                         0.130679   \n",
       "5                         0.098415                         0.127919   \n",
       "6                         0.192280                         0.201744   \n",
       "7                         0.154649                         0.153193   \n",
       "8                         0.065078                         0.130754   \n",
       "9                         0.185571                         0.194345   \n",
       "\n",
       "   ucseq_ucseq_merged_SRR6425992_2  ucseq_ucseq_merged_SRR6425993_2  \\\n",
       "0                         0.114476                         0.128712   \n",
       "1                         0.105495                         0.065635   \n",
       "2                         0.112692                         0.130679   \n",
       "3                         0.000000                         0.104296   \n",
       "4                         0.104296                         0.000000   \n",
       "5                         0.098442                         0.097492   \n",
       "6                         0.185979                         0.188330   \n",
       "7                         0.151898                         0.150890   \n",
       "8                         0.104607                         0.017648   \n",
       "9                         0.187617                         0.181538   \n",
       "\n",
       "   ucseq_ucseq_merged_SRR6425995_2  ucseq_ucseq_merged_SRR6425997_2  \\\n",
       "0                         0.121284                         0.192597   \n",
       "1                         0.098415                         0.192280   \n",
       "2                         0.127919                         0.201744   \n",
       "3                         0.098442                         0.185979   \n",
       "4                         0.097492                         0.188330   \n",
       "5                         0.000000                         0.188214   \n",
       "6                         0.188214                         0.000000   \n",
       "7                         0.148225                         0.140543   \n",
       "8                         0.097711                         0.189119   \n",
       "9                         0.187657                         0.133573   \n",
       "\n",
       "   ucseq_ucseq_merged_SRR6425998_2  ucseq_ucseq_merged_SRR6425999_2  \\\n",
       "0                         0.133703                         0.130147   \n",
       "1                         0.154649                         0.065078   \n",
       "2                         0.153193                         0.130754   \n",
       "3                         0.151898                         0.104607   \n",
       "4                         0.150890                         0.017648   \n",
       "5                         0.148225                         0.097711   \n",
       "6                         0.140543                         0.189119   \n",
       "7                         0.000000                         0.151709   \n",
       "8                         0.151709                         0.000000   \n",
       "9                         0.173495                         0.189365   \n",
       "\n",
       "   ucseq_ucseq_merged_SRR6426000_2  ucseq_ucseq_merged_SRR6426001_2  \\\n",
       "0                         0.190308                         0.157289   \n",
       "1                         0.185571                         0.152304   \n",
       "2                         0.194345                         0.160089   \n",
       "3                         0.187617                         0.156525   \n",
       "4                         0.181538                         0.144520   \n",
       "5                         0.187657                         0.148615   \n",
       "6                         0.133573                         0.147395   \n",
       "7                         0.173495                         0.015556   \n",
       "8                         0.189365                         0.142723   \n",
       "9                         0.000000                         0.178485   \n",
       "\n",
       "   ucseq_ucseq_merged_SRR6426002_2  ucseq_ucseq_merged_SRR6426003_2  \\\n",
       "0                         0.131612                         0.182641   \n",
       "1                         0.068205                         0.175759   \n",
       "2                         0.134019                         0.186412   \n",
       "3                         0.106157                         0.180429   \n",
       "4                         0.021502                         0.174222   \n",
       "5                         0.099079                         0.175503   \n",
       "6                         0.182973                         0.188911   \n",
       "7                         0.150087                         0.176831   \n",
       "8                         0.019900                         0.173301   \n",
       "9                         0.150801                         0.156946   \n",
       "\n",
       "   ucseq_ucseq_merged_SRR6426004_2  \n",
       "0                         0.135230  \n",
       "1                         0.063171  \n",
       "2                         0.136197  \n",
       "3                         0.108032  \n",
       "4                         0.065990  \n",
       "5                         0.101917  \n",
       "6                         0.191627  \n",
       "7                         0.156294  \n",
       "8                         0.065265  \n",
       "9                         0.186393  "
      ]
     },
     "execution_count": 340,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.melt(df, id_vars=['sample'], value_vars= list(df.columns[1:]) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425990_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.118984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425991_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.067789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425992_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.114476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425993_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.128712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425995_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.121284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425997_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.192597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425998_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.133703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425999_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.130147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426000_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.190308</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            sample                         variable     value\n",
       "0  ucseq_ucseq_merged_SRR6425989_2  ucseq_ucseq_merged_SRR6425989_2  0.000000\n",
       "1  ucseq_ucseq_merged_SRR6425990_2  ucseq_ucseq_merged_SRR6425989_2  0.118984\n",
       "2  ucseq_ucseq_merged_SRR6425991_2  ucseq_ucseq_merged_SRR6425989_2  0.067789\n",
       "3  ucseq_ucseq_merged_SRR6425992_2  ucseq_ucseq_merged_SRR6425989_2  0.114476\n",
       "4  ucseq_ucseq_merged_SRR6425993_2  ucseq_ucseq_merged_SRR6425989_2  0.128712\n",
       "5  ucseq_ucseq_merged_SRR6425995_2  ucseq_ucseq_merged_SRR6425989_2  0.121284\n",
       "6  ucseq_ucseq_merged_SRR6425997_2  ucseq_ucseq_merged_SRR6425989_2  0.192597\n",
       "7  ucseq_ucseq_merged_SRR6425998_2  ucseq_ucseq_merged_SRR6425989_2  0.133703\n",
       "8  ucseq_ucseq_merged_SRR6425999_2  ucseq_ucseq_merged_SRR6425989_2  0.130147\n",
       "9  ucseq_ucseq_merged_SRR6426000_2  ucseq_ucseq_merged_SRR6425989_2  0.190308"
      ]
     },
     "execution_count": 342,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 344,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "big_df = pd.DataFrame()\n",
    "\n",
    "for f in file_lst[:]:\n",
    "    finame = f.split(\"/\")[1]\n",
    "    \n",
    "    df = pd.read_csv(f, header = 0, sep='\\t', skiprows = 0)\n",
    "    df=pd.melt(df, id_vars=['sample'], value_vars= list(df.columns[1:]) )\n",
    "    \n",
    "    if \"rep\" in finame:\n",
    "        rep_num = float(re.split('[.|rep]', finame)[-2])\n",
    "        #print(rep_num)\n",
    "        df['cond'] = \"rep\"\n",
    "        df['rep_num'] = rep_num\n",
    "        \n",
    "    if \"main\" in finame:\n",
    "        df['cond'] = \"main\"\n",
    "        df['rep_num'] = float(-1)\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "    if f == file_lst[0]:\n",
    "        big_df = df\n",
    "        \n",
    "    else:\n",
    "        big_df = big_df.append(df)\n",
    "        \n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"di_mtrx/dimtrx_main.txt\", header = 0, sep='\\t', skiprows = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.melt(df, id_vars=['sample'], value_vars= list(df.columns[1:]) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.merge(big_df, df,  how='left', left_on=['sample','variable'], right_on = ['sample','variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>value_x</th>\n",
       "      <th>cond</th>\n",
       "      <th>rep_num</th>\n",
       "      <th>value_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425990_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.118984</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.118984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425991_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.067789</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.067789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425992_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.114476</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.114476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425993_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.128712</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.128712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425995_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.121284</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.121284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425997_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.192597</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.192597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425998_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.133703</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.133703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425999_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.130147</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.130147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426000_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.190308</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.190308</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            sample                         variable   value_x  \\\n",
       "0  ucseq_ucseq_merged_SRR6425989_2  ucseq_ucseq_merged_SRR6425989_2  0.000000   \n",
       "1  ucseq_ucseq_merged_SRR6425990_2  ucseq_ucseq_merged_SRR6425989_2  0.118984   \n",
       "2  ucseq_ucseq_merged_SRR6425991_2  ucseq_ucseq_merged_SRR6425989_2  0.067789   \n",
       "3  ucseq_ucseq_merged_SRR6425992_2  ucseq_ucseq_merged_SRR6425989_2  0.114476   \n",
       "4  ucseq_ucseq_merged_SRR6425993_2  ucseq_ucseq_merged_SRR6425989_2  0.128712   \n",
       "5  ucseq_ucseq_merged_SRR6425995_2  ucseq_ucseq_merged_SRR6425989_2  0.121284   \n",
       "6  ucseq_ucseq_merged_SRR6425997_2  ucseq_ucseq_merged_SRR6425989_2  0.192597   \n",
       "7  ucseq_ucseq_merged_SRR6425998_2  ucseq_ucseq_merged_SRR6425989_2  0.133703   \n",
       "8  ucseq_ucseq_merged_SRR6425999_2  ucseq_ucseq_merged_SRR6425989_2  0.130147   \n",
       "9  ucseq_ucseq_merged_SRR6426000_2  ucseq_ucseq_merged_SRR6425989_2  0.190308   \n",
       "\n",
       "   cond  rep_num   value_y  \n",
       "0  main     -1.0  0.000000  \n",
       "1  main     -1.0  0.118984  \n",
       "2  main     -1.0  0.067789  \n",
       "3  main     -1.0  0.114476  \n",
       "4  main     -1.0  0.128712  \n",
       "5  main     -1.0  0.121284  \n",
       "6  main     -1.0  0.192597  \n",
       "7  main     -1.0  0.133703  \n",
       "8  main     -1.0  0.130147  \n",
       "9  main     -1.0  0.190308  "
      ]
     },
     "execution_count": 348,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>value_x</th>\n",
       "      <th>cond</th>\n",
       "      <th>rep_num</th>\n",
       "      <th>value_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>rep</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425990_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.141288</td>\n",
       "      <td>rep</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.118984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425991_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.090150</td>\n",
       "      <td>rep</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.067789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425992_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.135174</td>\n",
       "      <td>rep</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.114476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425993_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.148306</td>\n",
       "      <td>rep</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.128712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19791</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426000_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.205673</td>\n",
       "      <td>rep</td>\n",
       "      <td>99.0</td>\n",
       "      <td>0.186393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19792</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426001_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.171855</td>\n",
       "      <td>rep</td>\n",
       "      <td>99.0</td>\n",
       "      <td>0.153240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19793</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426002_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.084835</td>\n",
       "      <td>rep</td>\n",
       "      <td>99.0</td>\n",
       "      <td>0.068335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19794</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426003_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.152700</td>\n",
       "      <td>rep</td>\n",
       "      <td>99.0</td>\n",
       "      <td>0.129711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19795</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>rep</td>\n",
       "      <td>99.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19600 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                sample                         variable  \\\n",
       "196    ucseq_ucseq_merged_SRR6425989_2  ucseq_ucseq_merged_SRR6425989_2   \n",
       "197    ucseq_ucseq_merged_SRR6425990_2  ucseq_ucseq_merged_SRR6425989_2   \n",
       "198    ucseq_ucseq_merged_SRR6425991_2  ucseq_ucseq_merged_SRR6425989_2   \n",
       "199    ucseq_ucseq_merged_SRR6425992_2  ucseq_ucseq_merged_SRR6425989_2   \n",
       "200    ucseq_ucseq_merged_SRR6425993_2  ucseq_ucseq_merged_SRR6425989_2   \n",
       "...                                ...                              ...   \n",
       "19791  ucseq_ucseq_merged_SRR6426000_2  ucseq_ucseq_merged_SRR6426004_2   \n",
       "19792  ucseq_ucseq_merged_SRR6426001_2  ucseq_ucseq_merged_SRR6426004_2   \n",
       "19793  ucseq_ucseq_merged_SRR6426002_2  ucseq_ucseq_merged_SRR6426004_2   \n",
       "19794  ucseq_ucseq_merged_SRR6426003_2  ucseq_ucseq_merged_SRR6426004_2   \n",
       "19795  ucseq_ucseq_merged_SRR6426004_2  ucseq_ucseq_merged_SRR6426004_2   \n",
       "\n",
       "        value_x cond  rep_num   value_y  \n",
       "196    0.000000  rep      0.0  0.000000  \n",
       "197    0.141288  rep      0.0  0.118984  \n",
       "198    0.090150  rep      0.0  0.067789  \n",
       "199    0.135174  rep      0.0  0.114476  \n",
       "200    0.148306  rep      0.0  0.128712  \n",
       "...         ...  ...      ...       ...  \n",
       "19791  0.205673  rep     99.0  0.186393  \n",
       "19792  0.171855  rep     99.0  0.153240  \n",
       "19793  0.084835  rep     99.0  0.068335  \n",
       "19794  0.152700  rep     99.0  0.129711  \n",
       "19795  0.000000  rep     99.0  0.000000  \n",
       "\n",
       "[19600 rows x 6 columns]"
      ]
     },
     "execution_count": 351,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "metadata": {},
   "outputs": [],
   "source": [
    "# compute centering\n",
    "d1 = new_df.drop(new_df[new_df.rep_num==-1].index)\n",
    "res = d1.groupby(['sample', 'variable'], as_index=False)['value_x'].mean()\n",
    "res.rename({'value_x': 'mean'}, axis=1, inplace=True)\n",
    "new_dff = pd.merge(new_df, res,  how='left', left_on=['sample','variable'], right_on = ['sample','variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 355,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>value_x</th>\n",
       "      <th>cond</th>\n",
       "      <th>rep_num</th>\n",
       "      <th>value_y</th>\n",
       "      <th>mean</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425990_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.118984</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.118984</td>\n",
       "      <td>0.140796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425991_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.067789</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.067789</td>\n",
       "      <td>0.090655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425992_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.114476</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.114476</td>\n",
       "      <td>0.135822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425993_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.128712</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.128712</td>\n",
       "      <td>0.148398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425995_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.121284</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.121284</td>\n",
       "      <td>0.142523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425997_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.192597</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.192597</td>\n",
       "      <td>0.217342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425998_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.133703</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.133703</td>\n",
       "      <td>0.158276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425999_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.130147</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.130147</td>\n",
       "      <td>0.149383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426000_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.190308</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.190308</td>\n",
       "      <td>0.214123</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            sample                         variable   value_x  \\\n",
       "0  ucseq_ucseq_merged_SRR6425989_2  ucseq_ucseq_merged_SRR6425989_2  0.000000   \n",
       "1  ucseq_ucseq_merged_SRR6425990_2  ucseq_ucseq_merged_SRR6425989_2  0.118984   \n",
       "2  ucseq_ucseq_merged_SRR6425991_2  ucseq_ucseq_merged_SRR6425989_2  0.067789   \n",
       "3  ucseq_ucseq_merged_SRR6425992_2  ucseq_ucseq_merged_SRR6425989_2  0.114476   \n",
       "4  ucseq_ucseq_merged_SRR6425993_2  ucseq_ucseq_merged_SRR6425989_2  0.128712   \n",
       "5  ucseq_ucseq_merged_SRR6425995_2  ucseq_ucseq_merged_SRR6425989_2  0.121284   \n",
       "6  ucseq_ucseq_merged_SRR6425997_2  ucseq_ucseq_merged_SRR6425989_2  0.192597   \n",
       "7  ucseq_ucseq_merged_SRR6425998_2  ucseq_ucseq_merged_SRR6425989_2  0.133703   \n",
       "8  ucseq_ucseq_merged_SRR6425999_2  ucseq_ucseq_merged_SRR6425989_2  0.130147   \n",
       "9  ucseq_ucseq_merged_SRR6426000_2  ucseq_ucseq_merged_SRR6425989_2  0.190308   \n",
       "\n",
       "   cond  rep_num   value_y      mean  \n",
       "0  main     -1.0  0.000000  0.000000  \n",
       "1  main     -1.0  0.118984  0.140796  \n",
       "2  main     -1.0  0.067789  0.090655  \n",
       "3  main     -1.0  0.114476  0.135822  \n",
       "4  main     -1.0  0.128712  0.148398  \n",
       "5  main     -1.0  0.121284  0.142523  \n",
       "6  main     -1.0  0.192597  0.217342  \n",
       "7  main     -1.0  0.133703  0.158276  \n",
       "8  main     -1.0  0.130147  0.149383  \n",
       "9  main     -1.0  0.190308  0.214123  "
      ]
     },
     "execution_count": 355,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_dff.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_dff['est_true'] =  ((new_dff['value_x']-new_dff['mean'])+new_dff['value_y'])/new_dff['value_y']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 358,
   "metadata": {},
   "outputs": [],
   "source": [
    "#combine all files in the list\n",
    "#combined_csv = pd.concat([pd.read_csv(f) for f in all_filenames ])\n",
    "#export to csv\n",
    "new_dff.to_csv( \"combined_resample_all_dist2.csv\", index=False, encoding='utf-8-sig')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Subsample "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "metadata": {},
   "outputs": [],
   "source": [
    "report_directory = \"di_mtrx_subsample\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "file_lst = glob.glob(report_directory+\"/di*.txt\")\n",
    "file_lst.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(file_lst[0], header = None, sep='\\t', skiprows = 1, index_col=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 324,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 14 entries, 0 to 13\n",
      "Data columns (total 15 columns):\n",
      " #   Column  Non-Null Count  Dtype  \n",
      "---  ------  --------------  -----  \n",
      " 0   0       14 non-null     object \n",
      " 1   1       14 non-null     float64\n",
      " 2   2       14 non-null     float64\n",
      " 3   3       14 non-null     float64\n",
      " 4   4       14 non-null     float64\n",
      " 5   5       14 non-null     float64\n",
      " 6   6       14 non-null     float64\n",
      " 7   7       14 non-null     float64\n",
      " 8   8       14 non-null     float64\n",
      " 9   9       14 non-null     float64\n",
      " 10  10      14 non-null     float64\n",
      " 11  11      14 non-null     float64\n",
      " 12  12      14 non-null     float64\n",
      " 13  13      14 non-null     float64\n",
      " 14  14      14 non-null     float64\n",
      "dtypes: float64(14), object(1)\n",
      "memory usage: 1.8+ KB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "metadata": {},
   "outputs": [],
   "source": [
    "big_df = pd.DataFrame()\n",
    "\n",
    "for f in file_lst[:]:\n",
    "    finame = f.split(\"/\")[1]\n",
    "    \n",
    "    if \"rep\" in finame:\n",
    "        df = pd.read_csv(f, header = None, sep='\\t', skiprows = 1, )\n",
    "    else:\n",
    "        df = pd.read_csv(f, header = 0, sep='\\t', skiprows = 0)\n",
    "    \n",
    "    \n",
    "    if \"rep\" in finame:\n",
    "        new_columns = list(df.iloc[:, 0])\n",
    "        new_columns.insert(0, \"sample\")\n",
    "        df.columns = new_columns\n",
    "    \n",
    "    df=pd.melt(df, id_vars=['sample'], value_vars= list(df.columns[1:]) )\n",
    "    \n",
    "    if \"rep\" in finame:\n",
    "        rep_num = float(re.split('[.|rep]', finame)[-2])\n",
    "        #print(rep_num)\n",
    "        df['cond'] = \"rep\"\n",
    "        df['rep_num'] = rep_num\n",
    "        \n",
    "    if \"main\" in finame:\n",
    "        df['cond'] = \"main\"\n",
    "        df['rep_num'] = float(-1)\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "    if f == file_lst[0]:\n",
    "        big_df = df\n",
    "        \n",
    "    else:\n",
    "        big_df = big_df.append(df)\n",
    "        \n",
    "        \n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"di_mtrx/dimtrx_main.txt\", header = 0, sep='\\t', skiprows = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.melt(df, id_vars=['sample'], value_vars= list(df.columns[1:]) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.merge(big_df, df,  how='left', left_on=['sample','variable'], right_on = ['sample','variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>value_x</th>\n",
       "      <th>cond</th>\n",
       "      <th>rep_num</th>\n",
       "      <th>value_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>rep</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425990_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.120143</td>\n",
       "      <td>rep</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.118984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425991_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.068039</td>\n",
       "      <td>rep</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.067789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425992_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.112919</td>\n",
       "      <td>rep</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.114476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6425993_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6425989_2</td>\n",
       "      <td>0.129562</td>\n",
       "      <td>rep</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.128712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19791</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426000_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.186393</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.186393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19792</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426001_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.153240</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.153240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19793</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426002_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.068335</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.068335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19794</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426003_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.129711</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.129711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19795</th>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>ucseq_ucseq_merged_SRR6426004_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>main</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19796 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                sample                         variable  \\\n",
       "0      ucseq_ucseq_merged_SRR6425989_2  ucseq_ucseq_merged_SRR6425989_2   \n",
       "1      ucseq_ucseq_merged_SRR6425990_2  ucseq_ucseq_merged_SRR6425989_2   \n",
       "2      ucseq_ucseq_merged_SRR6425991_2  ucseq_ucseq_merged_SRR6425989_2   \n",
       "3      ucseq_ucseq_merged_SRR6425992_2  ucseq_ucseq_merged_SRR6425989_2   \n",
       "4      ucseq_ucseq_merged_SRR6425993_2  ucseq_ucseq_merged_SRR6425989_2   \n",
       "...                                ...                              ...   \n",
       "19791  ucseq_ucseq_merged_SRR6426000_2  ucseq_ucseq_merged_SRR6426004_2   \n",
       "19792  ucseq_ucseq_merged_SRR6426001_2  ucseq_ucseq_merged_SRR6426004_2   \n",
       "19793  ucseq_ucseq_merged_SRR6426002_2  ucseq_ucseq_merged_SRR6426004_2   \n",
       "19794  ucseq_ucseq_merged_SRR6426003_2  ucseq_ucseq_merged_SRR6426004_2   \n",
       "19795  ucseq_ucseq_merged_SRR6426004_2  ucseq_ucseq_merged_SRR6426004_2   \n",
       "\n",
       "        value_x  cond  rep_num   value_y  \n",
       "0      0.000000   rep      0.0  0.000000  \n",
       "1      0.120143   rep      0.0  0.118984  \n",
       "2      0.068039   rep      0.0  0.067789  \n",
       "3      0.112919   rep      0.0  0.114476  \n",
       "4      0.129562   rep      0.0  0.128712  \n",
       "...         ...   ...      ...       ...  \n",
       "19791  0.186393  main     -1.0  0.186393  \n",
       "19792  0.153240  main     -1.0  0.153240  \n",
       "19793  0.068335  main     -1.0  0.068335  \n",
       "19794  0.129711  main     -1.0  0.129711  \n",
       "19795  0.000000  main     -1.0  0.000000  \n",
       "\n",
       "[19796 rows x 6 columns]"
      ]
     },
     "execution_count": 329,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 330,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df['est_true'] =  new_df['value_x']/new_df['value_y']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "metadata": {},
   "outputs": [],
   "source": [
    "#combine all files in the list\n",
    "#combined_csv = pd.concat([pd.read_csv(f) for f in all_filenames ])\n",
    "#export to csv\n",
    "new_df.to_csv( \"combined_subsample_all_dist.csv\", index=False, encoding='utf-8-sig')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# end of the notebook"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.rename(columns={'value':'dist'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "#sp_names = set(df[\"sample\"].to_list())\n",
    "sp_names = set(df.iloc[:, 0].to_list())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'ucseq_ucseq_merged_SRR6425989_2', 'ucseq_ucseq_merged_SRR6426004_2', 'ucseq_ucseq_merged_SRR6426000_2', 'ucseq_ucseq_merged_SRR6426003_2', 'ucseq_ucseq_merged_SRR6425999_2', 'ucseq_ucseq_merged_SRR6426002_2', 'ucseq_ucseq_merged_SRR6425998_2', 'ucseq_ucseq_merged_SRR6425991_2', 'ucseq_ucseq_merged_SRR6425993_2', 'ucseq_ucseq_merged_SRR6425997_2', 'ucseq_ucseq_merged_SRR6426001_2', 'ucseq_ucseq_merged_SRR6425992_2', 'ucseq_ucseq_merged_SRR6425990_2', 'ucseq_ucseq_merged_SRR6425995_2'}\n"
     ]
    }
   ],
   "source": [
    "print(sp_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ucseq_ucseq_merged_SRR6425989_2\n",
      "ucseq_ucseq_merged_SRR6426004_2\n",
      "ucseq_ucseq_merged_SRR6426000_2\n",
      "ucseq_ucseq_merged_SRR6426003_2\n",
      "ucseq_ucseq_merged_SRR6425999_2\n",
      "ucseq_ucseq_merged_SRR6426002_2\n",
      "ucseq_ucseq_merged_SRR6425998_2\n",
      "ucseq_ucseq_merged_SRR6425991_2\n",
      "ucseq_ucseq_merged_SRR6425993_2\n",
      "ucseq_ucseq_merged_SRR6425997_2\n",
      "ucseq_ucseq_merged_SRR6426001_2\n",
      "ucseq_ucseq_merged_SRR6425992_2\n",
      "ucseq_ucseq_merged_SRR6425990_2\n",
      "ucseq_ucseq_merged_SRR6425995_2\n",
      "{'ucseq_ucseq_merged_SRR6425989_2': 'merged_SRR6425989', 'ucseq_ucseq_merged_SRR6426004_2': 'merged_SRR6426004', 'ucseq_ucseq_merged_SRR6426000_2': 'merged_SRR6426000', 'ucseq_ucseq_merged_SRR6426003_2': 'merged_SRR6426003', 'ucseq_ucseq_merged_SRR6425999_2': 'merged_SRR6425999', 'ucseq_ucseq_merged_SRR6426002_2': 'merged_SRR6426002', 'ucseq_ucseq_merged_SRR6425998_2': 'merged_SRR6425998', 'ucseq_ucseq_merged_SRR6425991_2': 'merged_SRR6425991', 'ucseq_ucseq_merged_SRR6425993_2': 'merged_SRR6425993', 'ucseq_ucseq_merged_SRR6425997_2': 'merged_SRR6425997', 'ucseq_ucseq_merged_SRR6426001_2': 'merged_SRR6426001', 'ucseq_ucseq_merged_SRR6425992_2': 'merged_SRR6425992', 'ucseq_ucseq_merged_SRR6425990_2': 'merged_SRR6425990', 'ucseq_ucseq_merged_SRR6425995_2': 'merged_SRR6425995'}\n"
     ]
    }
   ],
   "source": [
    "import csv\n",
    "#filename =open(\"dimat_simPrimates.csv\", mode='r',encoding='utf8')\n",
    "fname_map = {}\n",
    "\n",
    "#sp_map = pd.read_csv(filename, index_col=0, skiprows = 0).T.to_dict()\n",
    "\n",
    "for k in sp_names:\n",
    "    print(k)\n",
    "    \n",
    "    #print(k.split(\".\")[0])\n",
    "    #new_k = k.split(\".\")[0]\n",
    "    #new_k = k + \"_1\"\n",
    "    #new_k = \"sub_{}_2\".format(sp_map[k][\"sp\"])\n",
    "    \n",
    "    \n",
    "    fname_map[k] = k.split(\"_\")[-3] + \"_\" + k.split(\"_\")[-2]\n",
    "    #fname_map[k] = \"sim\"+k.split(\"_\")[0] #(4000 genes, nucleotide simulation)\n",
    "    \n",
    "print(fname_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.replace({'sample': fname_map}, inplace = True)\n",
    "df.replace({'variable': fname_map}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "#new_df = pd.merge(new_df, df_lsh,  how='left', left_on=['sample','variable'], right_on = ['sample','variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "my_newick_string = glob.glob(report_directory+\"/*.nwk\")[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "print(my_newick_string)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "import treeswift\n",
    "tree = treeswift.read_tree_newick(my_newick_string)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "print(tree)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "for node in tree.traverse_postorder():\n",
    "    print(node)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "map_leaves = tree.label_to_node(selection='all')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "M = tree.distance_matrix()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "print(M)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "#d = tree.distance_between(map_leaves['n3'],map_leaves['n4'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df['dist'] = 0.0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "for index, row in df.iterrows():\n",
    "    df.at[index, 'dist'] = tree.distance_between(map_leaves[row['sample']], map_leaves[row['variable']])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df.head(n=10)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "df['cond'] = \"true_dist\"\n",
    "#df['test'] = \"true\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "arr = fname_map.values()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "r = 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "from itertools import combinations\n",
    "combos = list(combinations(arr, r))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ast import literal_eval as make_tuple"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "for index in range (0, len(df)):\n",
    "    print(df.loc[index, 'sample'], df.loc[index, 'variable'])\n",
    "    a = df.loc[index, 'sample']\n",
    "    b = df.loc[index, 'variable']\n",
    "    res = \"(\" + \"'\" + str(a) + \"'\" + \", \" + \"'\" + str(b) + \"'\" + \")\"\n",
    "    #print(res)\n",
    "    #print(make_tuple(res))\n",
    "    \n",
    "    if make_tuple(res) in combos:\n",
    "        print(\"here\")\n",
    "        df.loc[index, 'status'] = 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df.head(n=10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df = df.drop(df[df.status == 0].index)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df.head(n=10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "true_dist_const = df.iloc[0][\"dist\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "import os\n",
    "path = os.path.abspath(os.getcwd()) + \"/\"+ report_directory\n",
    "print(path)\n",
    "df.to_csv(os.path.join(path, r\"true_dist.csv\"), index = None)\n",
    "\n",
    "#export_csv = new_df.to_csv(r\"1x_simPrimetes_param.csv\", index = None, header=True)\n",
    "\n",
    "#export_csv = new_df.to_csv(r\"0.1x_simPrimetes_param.csv\", index = None, header=True)\n",
    "#export_csv = new_df.to_csv(r\"1x_fixed_simPrimetes_param.csv\", index = None, header=True)\n",
    "#export_csv = new_df.to_csv(r\"fixed_simPrimetes_nuc_dist.csv\", index = None, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "big_df = df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ucseq_ucseq_sub_Drosophila_ananassae_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.119075</td>\n",
       "      <td>0.068139</td>\n",
       "      <td>0.137842</td>\n",
       "      <td>0.123039</td>\n",
       "      <td>0.136521</td>\n",
       "      <td>0.196069</td>\n",
       "      <td>0.158644</td>\n",
       "      <td>0.162455</td>\n",
       "      <td>0.137949</td>\n",
       "      <td>0.135770</td>\n",
       "      <td>0.190288</td>\n",
       "      <td>0.185712</td>\n",
       "      <td>0.138508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ucseq_ucseq_sub_Drosophila_biarmipes_2</td>\n",
       "      <td>0.119075</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.116601</td>\n",
       "      <td>0.107105</td>\n",
       "      <td>0.097306</td>\n",
       "      <td>0.104457</td>\n",
       "      <td>0.193020</td>\n",
       "      <td>0.157478</td>\n",
       "      <td>0.156559</td>\n",
       "      <td>0.107470</td>\n",
       "      <td>0.103532</td>\n",
       "      <td>0.187293</td>\n",
       "      <td>0.183553</td>\n",
       "      <td>0.108378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ucseq_ucseq_sub_Drosophila_bipectinata_2</td>\n",
       "      <td>0.068139</td>\n",
       "      <td>0.116601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.126580</td>\n",
       "      <td>0.116011</td>\n",
       "      <td>0.131394</td>\n",
       "      <td>0.193197</td>\n",
       "      <td>0.140114</td>\n",
       "      <td>0.158482</td>\n",
       "      <td>0.133704</td>\n",
       "      <td>0.131040</td>\n",
       "      <td>0.180952</td>\n",
       "      <td>0.180910</td>\n",
       "      <td>0.134359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ucseq_ucseq_sub_Drosophila_erecta_2</td>\n",
       "      <td>0.137842</td>\n",
       "      <td>0.107105</td>\n",
       "      <td>0.126580</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.101785</td>\n",
       "      <td>0.065506</td>\n",
       "      <td>0.192916</td>\n",
       "      <td>0.159242</td>\n",
       "      <td>0.157388</td>\n",
       "      <td>0.068806</td>\n",
       "      <td>0.064806</td>\n",
       "      <td>0.188196</td>\n",
       "      <td>0.181106</td>\n",
       "      <td>0.062645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>ucseq_ucseq_sub_Drosophila_eugracilis_2</td>\n",
       "      <td>0.123039</td>\n",
       "      <td>0.097306</td>\n",
       "      <td>0.116011</td>\n",
       "      <td>0.101785</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.099355</td>\n",
       "      <td>0.190223</td>\n",
       "      <td>0.151930</td>\n",
       "      <td>0.150990</td>\n",
       "      <td>0.102523</td>\n",
       "      <td>0.098702</td>\n",
       "      <td>0.185005</td>\n",
       "      <td>0.177985</td>\n",
       "      <td>0.103593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>ucseq_ucseq_sub_Drosophila_mauritiana_2</td>\n",
       "      <td>0.136521</td>\n",
       "      <td>0.104457</td>\n",
       "      <td>0.131394</td>\n",
       "      <td>0.065506</td>\n",
       "      <td>0.099355</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.189607</td>\n",
       "      <td>0.153960</td>\n",
       "      <td>0.149059</td>\n",
       "      <td>0.020485</td>\n",
       "      <td>0.016634</td>\n",
       "      <td>0.178562</td>\n",
       "      <td>0.177479</td>\n",
       "      <td>0.065090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>ucseq_ucseq_sub_Drosophila_mojavensis_2</td>\n",
       "      <td>0.196069</td>\n",
       "      <td>0.193020</td>\n",
       "      <td>0.193197</td>\n",
       "      <td>0.192916</td>\n",
       "      <td>0.190223</td>\n",
       "      <td>0.189607</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.156109</td>\n",
       "      <td>0.163210</td>\n",
       "      <td>0.188032</td>\n",
       "      <td>0.190619</td>\n",
       "      <td>0.136459</td>\n",
       "      <td>0.190113</td>\n",
       "      <td>0.193323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>ucseq_ucseq_sub_Drosophila_persimilis_2</td>\n",
       "      <td>0.158644</td>\n",
       "      <td>0.157478</td>\n",
       "      <td>0.140114</td>\n",
       "      <td>0.159242</td>\n",
       "      <td>0.151930</td>\n",
       "      <td>0.153960</td>\n",
       "      <td>0.156109</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.014277</td>\n",
       "      <td>0.155469</td>\n",
       "      <td>0.156014</td>\n",
       "      <td>0.173258</td>\n",
       "      <td>0.188543</td>\n",
       "      <td>0.158887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>ucseq_ucseq_sub_Drosophila_pseudoobscura_2</td>\n",
       "      <td>0.162455</td>\n",
       "      <td>0.156559</td>\n",
       "      <td>0.158482</td>\n",
       "      <td>0.157388</td>\n",
       "      <td>0.150990</td>\n",
       "      <td>0.149059</td>\n",
       "      <td>0.163210</td>\n",
       "      <td>0.014277</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.144450</td>\n",
       "      <td>0.148566</td>\n",
       "      <td>0.177948</td>\n",
       "      <td>0.189802</td>\n",
       "      <td>0.156470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>ucseq_ucseq_sub_Drosophila_sechellia_2</td>\n",
       "      <td>0.137949</td>\n",
       "      <td>0.107470</td>\n",
       "      <td>0.133704</td>\n",
       "      <td>0.068806</td>\n",
       "      <td>0.102523</td>\n",
       "      <td>0.020485</td>\n",
       "      <td>0.188032</td>\n",
       "      <td>0.155469</td>\n",
       "      <td>0.144450</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.018888</td>\n",
       "      <td>0.166471</td>\n",
       "      <td>0.173289</td>\n",
       "      <td>0.067937</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           0         1         2         3   \\\n",
       "0      ucseq_ucseq_sub_Drosophila_ananassae_2  0.000000  0.119075  0.068139   \n",
       "1      ucseq_ucseq_sub_Drosophila_biarmipes_2  0.119075  0.000000  0.116601   \n",
       "2    ucseq_ucseq_sub_Drosophila_bipectinata_2  0.068139  0.116601  0.000000   \n",
       "3         ucseq_ucseq_sub_Drosophila_erecta_2  0.137842  0.107105  0.126580   \n",
       "4     ucseq_ucseq_sub_Drosophila_eugracilis_2  0.123039  0.097306  0.116011   \n",
       "5     ucseq_ucseq_sub_Drosophila_mauritiana_2  0.136521  0.104457  0.131394   \n",
       "6     ucseq_ucseq_sub_Drosophila_mojavensis_2  0.196069  0.193020  0.193197   \n",
       "7     ucseq_ucseq_sub_Drosophila_persimilis_2  0.158644  0.157478  0.140114   \n",
       "8  ucseq_ucseq_sub_Drosophila_pseudoobscura_2  0.162455  0.156559  0.158482   \n",
       "9      ucseq_ucseq_sub_Drosophila_sechellia_2  0.137949  0.107470  0.133704   \n",
       "\n",
       "         4         5         6         7         8         9         10  \\\n",
       "0  0.137842  0.123039  0.136521  0.196069  0.158644  0.162455  0.137949   \n",
       "1  0.107105  0.097306  0.104457  0.193020  0.157478  0.156559  0.107470   \n",
       "2  0.126580  0.116011  0.131394  0.193197  0.140114  0.158482  0.133704   \n",
       "3  0.000000  0.101785  0.065506  0.192916  0.159242  0.157388  0.068806   \n",
       "4  0.101785  0.000000  0.099355  0.190223  0.151930  0.150990  0.102523   \n",
       "5  0.065506  0.099355  0.000000  0.189607  0.153960  0.149059  0.020485   \n",
       "6  0.192916  0.190223  0.189607  0.000000  0.156109  0.163210  0.188032   \n",
       "7  0.159242  0.151930  0.153960  0.156109  0.000000  0.014277  0.155469   \n",
       "8  0.157388  0.150990  0.149059  0.163210  0.014277  0.000000  0.144450   \n",
       "9  0.068806  0.102523  0.020485  0.188032  0.155469  0.144450  0.000000   \n",
       "\n",
       "         11        12        13        14  \n",
       "0  0.135770  0.190288  0.185712  0.138508  \n",
       "1  0.103532  0.187293  0.183553  0.108378  \n",
       "2  0.131040  0.180952  0.180910  0.134359  \n",
       "3  0.064806  0.188196  0.181106  0.062645  \n",
       "4  0.098702  0.185005  0.177985  0.103593  \n",
       "5  0.016634  0.178562  0.177479  0.065090  \n",
       "6  0.190619  0.136459  0.190113  0.193323  \n",
       "7  0.156014  0.173258  0.188543  0.158887  \n",
       "8  0.148566  0.177948  0.189802  0.156470  \n",
       "9  0.018888  0.166471  0.173289  0.067937  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "big_df.head(n=10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "def translate(txt):\n",
    "    return txt.split(\"_\")[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "JC correction"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "import math\n",
    "def jc(b, do_correction, a):\n",
    "    deps = 0.0001\n",
    "    if abs(b) < deps:\n",
    "        return 0\n",
    "    loc = 1 - (4.0 * b / 3)\n",
    "\n",
    "    if (loc <= 0):\n",
    "        return 5.0 # max dist\n",
    "    if do_correction:\n",
    "        if a < deps:\n",
    "            a = deps\n",
    "        ret = a * (0.75 * (loc**(-1.0/a) - 1))\n",
    "    else:\n",
    "        ret = -0.75 * (math.log(loc))\n",
    "    return ret"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "do_correction = False; # \n",
    "alpha = 1.0;"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "#big_df['strap_iter'] = float('inf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "big_df = pd.DataFrame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "di_mtrx/dimtrx_no_strap.txt\n",
      "di_mtrx/dimtrx_strap_0.txt\n",
      "di_mtrx/dimtrx_strap_1.txt\n",
      "di_mtrx/dimtrx_strap_10.txt\n",
      "di_mtrx/dimtrx_strap_11.txt\n",
      "di_mtrx/dimtrx_strap_12.txt\n",
      "di_mtrx/dimtrx_strap_13.txt\n",
      "di_mtrx/dimtrx_strap_14.txt\n",
      "di_mtrx/dimtrx_strap_15.txt\n",
      "di_mtrx/dimtrx_strap_16.txt\n",
      "di_mtrx/dimtrx_strap_17.txt\n",
      "di_mtrx/dimtrx_strap_18.txt\n",
      "di_mtrx/dimtrx_strap_19.txt\n",
      "di_mtrx/dimtrx_strap_2.txt\n",
      "di_mtrx/dimtrx_strap_20.txt\n",
      "di_mtrx/dimtrx_strap_21.txt\n",
      "di_mtrx/dimtrx_strap_22.txt\n",
      "di_mtrx/dimtrx_strap_23.txt\n",
      "di_mtrx/dimtrx_strap_24.txt\n",
      "di_mtrx/dimtrx_strap_25.txt\n",
      "di_mtrx/dimtrx_strap_26.txt\n",
      "di_mtrx/dimtrx_strap_27.txt\n",
      "di_mtrx/dimtrx_strap_28.txt\n",
      "di_mtrx/dimtrx_strap_29.txt\n",
      "di_mtrx/dimtrx_strap_3.txt\n",
      "di_mtrx/dimtrx_strap_30.txt\n",
      "di_mtrx/dimtrx_strap_31.txt\n",
      "di_mtrx/dimtrx_strap_32.txt\n",
      "di_mtrx/dimtrx_strap_33.txt\n",
      "di_mtrx/dimtrx_strap_34.txt\n",
      "di_mtrx/dimtrx_strap_35.txt\n",
      "di_mtrx/dimtrx_strap_36.txt\n",
      "di_mtrx/dimtrx_strap_37.txt\n",
      "di_mtrx/dimtrx_strap_38.txt\n",
      "di_mtrx/dimtrx_strap_39.txt\n",
      "di_mtrx/dimtrx_strap_4.txt\n",
      "di_mtrx/dimtrx_strap_40.txt\n",
      "di_mtrx/dimtrx_strap_41.txt\n",
      "di_mtrx/dimtrx_strap_42.txt\n",
      "di_mtrx/dimtrx_strap_43.txt\n",
      "di_mtrx/dimtrx_strap_44.txt\n",
      "di_mtrx/dimtrx_strap_45.txt\n",
      "di_mtrx/dimtrx_strap_46.txt\n",
      "di_mtrx/dimtrx_strap_47.txt\n",
      "di_mtrx/dimtrx_strap_48.txt\n",
      "di_mtrx/dimtrx_strap_49.txt\n",
      "di_mtrx/dimtrx_strap_5.txt\n",
      "di_mtrx/dimtrx_strap_50.txt\n",
      "di_mtrx/dimtrx_strap_51.txt\n",
      "di_mtrx/dimtrx_strap_52.txt\n",
      "di_mtrx/dimtrx_strap_53.txt\n",
      "di_mtrx/dimtrx_strap_54.txt\n",
      "di_mtrx/dimtrx_strap_55.txt\n",
      "di_mtrx/dimtrx_strap_56.txt\n",
      "di_mtrx/dimtrx_strap_57.txt\n",
      "di_mtrx/dimtrx_strap_58.txt\n",
      "di_mtrx/dimtrx_strap_59.txt\n",
      "di_mtrx/dimtrx_strap_6.txt\n",
      "di_mtrx/dimtrx_strap_60.txt\n",
      "di_mtrx/dimtrx_strap_61.txt\n",
      "di_mtrx/dimtrx_strap_62.txt\n",
      "di_mtrx/dimtrx_strap_63.txt\n",
      "di_mtrx/dimtrx_strap_64.txt\n",
      "di_mtrx/dimtrx_strap_65.txt\n",
      "di_mtrx/dimtrx_strap_66.txt\n",
      "di_mtrx/dimtrx_strap_67.txt\n",
      "di_mtrx/dimtrx_strap_68.txt\n",
      "di_mtrx/dimtrx_strap_69.txt\n",
      "di_mtrx/dimtrx_strap_7.txt\n",
      "di_mtrx/dimtrx_strap_70.txt\n",
      "di_mtrx/dimtrx_strap_71.txt\n",
      "di_mtrx/dimtrx_strap_72.txt\n",
      "di_mtrx/dimtrx_strap_73.txt\n",
      "di_mtrx/dimtrx_strap_74.txt\n",
      "di_mtrx/dimtrx_strap_75.txt\n",
      "di_mtrx/dimtrx_strap_76.txt\n",
      "di_mtrx/dimtrx_strap_77.txt\n",
      "di_mtrx/dimtrx_strap_78.txt\n",
      "di_mtrx/dimtrx_strap_79.txt\n",
      "di_mtrx/dimtrx_strap_8.txt\n",
      "di_mtrx/dimtrx_strap_80.txt\n",
      "di_mtrx/dimtrx_strap_81.txt\n",
      "di_mtrx/dimtrx_strap_82.txt\n",
      "di_mtrx/dimtrx_strap_83.txt\n",
      "di_mtrx/dimtrx_strap_84.txt\n",
      "di_mtrx/dimtrx_strap_85.txt\n",
      "di_mtrx/dimtrx_strap_86.txt\n",
      "di_mtrx/dimtrx_strap_87.txt\n",
      "di_mtrx/dimtrx_strap_88.txt\n",
      "di_mtrx/dimtrx_strap_89.txt\n",
      "di_mtrx/dimtrx_strap_9.txt\n",
      "di_mtrx/dimtrx_strap_90.txt\n",
      "di_mtrx/dimtrx_strap_91.txt\n",
      "di_mtrx/dimtrx_strap_92.txt\n",
      "di_mtrx/dimtrx_strap_93.txt\n",
      "di_mtrx/dimtrx_strap_94.txt\n",
      "di_mtrx/dimtrx_strap_95.txt\n",
      "di_mtrx/dimtrx_strap_96.txt\n",
      "di_mtrx/dimtrx_strap_97.txt\n",
      "di_mtrx/dimtrx_strap_98.txt\n",
      "di_mtrx/dimtrx_strap_99.txt\n"
     ]
    }
   ],
   "source": [
    "for f in file_lst[:]:\n",
    "    print(f)\n",
    "    \n",
    "    #df = pd.read_csv(f, header = 0, sep='\\t')\n",
    "    \n",
    "    df_tmp = pd.read_csv(f, header = None, sep='\\t', skiprows = 1)\n",
    "    headers = [\"sample\"]\n",
    "    headers.extend(df_tmp.iloc[:, 0].to_list())\n",
    "    \n",
    "    df = pd.read_csv(f, header = None, sep='\\t', skiprows = 1, names=headers)\n",
    "    \n",
    "    \n",
    "    df = pd.melt(df, id_vars=['sample'], value_vars= list(df.columns[1:]) )\n",
    "    #df[\"dist\"] = df[['value']].applymap(lambda x: jc(x, do_correction, alpha))\n",
    "    #df.drop(['value'], axis=1, inplace=True)\n",
    "    df.rename(columns={'value':'dist'}, inplace=True) # already corrected since run with skmer with -t\n",
    "    \n",
    "\n",
    "    \n",
    "    sp_names = set(df[\"sample\"].to_list())\n",
    "    fname_map = {}\n",
    "    for k in sp_names:\n",
    "        fname_map[k] = k.split(\"_\")[-3] + \"_\" + k.split(\"_\")[-2]\n",
    "    \n",
    "    \n",
    "    df.replace({'sample': fname_map}, inplace = True)\n",
    "    df.replace({'variable': fname_map}, inplace = True)\n",
    "    \n",
    "    #df['cond'] = \"expected_dist\"\n",
    "    #df['cond'] = report_directory\n",
    "    df['cond'] = \"rep\"\n",
    "    \n",
    "    \n",
    "    \n",
    "    if \"no_strap\" in f:\n",
    "        df['cond'] = \"est_dist\"\n",
    "   \n",
    "    \n",
    "    \n",
    "    #df['test'] = fname_map.keys()[0]\n",
    "    \n",
    "    df['status'] = 0\n",
    "    \n",
    "    for index in range (0, len(df)):\n",
    "        #print(df.loc[index, 'sample'], df.loc[index, 'variable'])\n",
    "        a = df.loc[index, 'sample']\n",
    "        b = df.loc[index, 'variable']\n",
    "        res = \"(\" + \"'\" + str(a) + \"'\" + \", \" + \"'\" + str(b) + \"'\" + \")\"\n",
    "        #print(res)\n",
    "        #print(make_tuple(res))\n",
    "    \n",
    "        if make_tuple(res) in combos:\n",
    "            df.loc[index, 'status'] = 1\n",
    "            \n",
    "    df = df.drop(df[df.status == 0].index)\n",
    "    \n",
    "    \n",
    "    # add strap number\n",
    "    df['strap_iter'] = float('inf')\n",
    "    \n",
    "    if \"no_strap\" in f:\n",
    "        df['strap_iter'] = - 1\n",
    "    else:\n",
    "        df['strap_iter'] = int([i for i in re.split('[/._]', f) if i.isnumeric()][-1])\n",
    "         \n",
    "    \n",
    "    if f == file_lst[0]:\n",
    "        big_df = df\n",
    "        \n",
    "    else:\n",
    "        big_df = big_df.append(df)\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>dist</th>\n",
       "      <th>cond</th>\n",
       "      <th>status</th>\n",
       "      <th>strap_iter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Drosophila_biarmipes</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.090728</td>\n",
       "      <td>rep</td>\n",
       "      <td>1</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Drosophila_eugracilis</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.088183</td>\n",
       "      <td>rep</td>\n",
       "      <td>1</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Drosophila_mauritiana</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.126098</td>\n",
       "      <td>rep</td>\n",
       "      <td>1</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Drosophila_persimilis</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.138634</td>\n",
       "      <td>rep</td>\n",
       "      <td>1</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Drosophila_simulans</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.128607</td>\n",
       "      <td>rep</td>\n",
       "      <td>1</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Drosophila_virilis</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.155244</td>\n",
       "      <td>rep</td>\n",
       "      <td>1</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Drosophila_yakuba</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.127756</td>\n",
       "      <td>rep</td>\n",
       "      <td>1</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Drosophila_eugracilis</td>\n",
       "      <td>Drosophila_biarmipes</td>\n",
       "      <td>0.065711</td>\n",
       "      <td>rep</td>\n",
       "      <td>1</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Drosophila_mauritiana</td>\n",
       "      <td>Drosophila_biarmipes</td>\n",
       "      <td>0.096926</td>\n",
       "      <td>rep</td>\n",
       "      <td>1</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Drosophila_simulans</td>\n",
       "      <td>Drosophila_biarmipes</td>\n",
       "      <td>0.098747</td>\n",
       "      <td>rep</td>\n",
       "      <td>1</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample              variable      dist cond  status  \\\n",
       "1    Drosophila_biarmipes  Drosophila_ananassae  0.090728  rep       1   \n",
       "4   Drosophila_eugracilis  Drosophila_ananassae  0.088183  rep       1   \n",
       "5   Drosophila_mauritiana  Drosophila_ananassae  0.126098  rep       1   \n",
       "7   Drosophila_persimilis  Drosophila_ananassae  0.138634  rep       1   \n",
       "10    Drosophila_simulans  Drosophila_ananassae  0.128607  rep       1   \n",
       "11     Drosophila_virilis  Drosophila_ananassae  0.155244  rep       1   \n",
       "13      Drosophila_yakuba  Drosophila_ananassae  0.127756  rep       1   \n",
       "18  Drosophila_eugracilis  Drosophila_biarmipes  0.065711  rep       1   \n",
       "19  Drosophila_mauritiana  Drosophila_biarmipes  0.096926  rep       1   \n",
       "24    Drosophila_simulans  Drosophila_biarmipes  0.098747  rep       1   \n",
       "\n",
       "    strap_iter  \n",
       "1           99  \n",
       "4           99  \n",
       "5           99  \n",
       "7           99  \n",
       "10          99  \n",
       "11          99  \n",
       "13          99  \n",
       "18          99  \n",
       "19          99  \n",
       "24          99  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>dist</th>\n",
       "      <th>cond</th>\n",
       "      <th>status</th>\n",
       "      <th>strap_iter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Drosophila_biarmipes</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.119075</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Drosophila_eugracilis</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.123039</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Drosophila_mauritiana</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.136521</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Drosophila_persimilis</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.158644</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Drosophila_simulans</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.135770</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Drosophila_virilis</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.190288</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Drosophila_yakuba</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.138508</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Drosophila_eugracilis</td>\n",
       "      <td>Drosophila_biarmipes</td>\n",
       "      <td>0.097306</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Drosophila_mauritiana</td>\n",
       "      <td>Drosophila_biarmipes</td>\n",
       "      <td>0.104457</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Drosophila_simulans</td>\n",
       "      <td>Drosophila_biarmipes</td>\n",
       "      <td>0.103532</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample              variable      dist      cond  status  \\\n",
       "1    Drosophila_biarmipes  Drosophila_ananassae  0.119075  est_dist       1   \n",
       "4   Drosophila_eugracilis  Drosophila_ananassae  0.123039  est_dist       1   \n",
       "5   Drosophila_mauritiana  Drosophila_ananassae  0.136521  est_dist       1   \n",
       "7   Drosophila_persimilis  Drosophila_ananassae  0.158644  est_dist       1   \n",
       "10    Drosophila_simulans  Drosophila_ananassae  0.135770  est_dist       1   \n",
       "11     Drosophila_virilis  Drosophila_ananassae  0.190288  est_dist       1   \n",
       "13      Drosophila_yakuba  Drosophila_ananassae  0.138508  est_dist       1   \n",
       "18  Drosophila_eugracilis  Drosophila_biarmipes  0.097306  est_dist       1   \n",
       "19  Drosophila_mauritiana  Drosophila_biarmipes  0.104457  est_dist       1   \n",
       "24    Drosophila_simulans  Drosophila_biarmipes  0.103532  est_dist       1   \n",
       "\n",
       "    strap_iter  \n",
       "1           -1  \n",
       "4           -1  \n",
       "5           -1  \n",
       "7           -1  \n",
       "10          -1  \n",
       "11          -1  \n",
       "13          -1  \n",
       "18          -1  \n",
       "19          -1  \n",
       "24          -1  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "big_df.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "#big_df['cond_tested'] = report_directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "#big_df['est_true'] = big_df['dist']/true_dist_const"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/admin/Documents/support/real_data/dros_p3c1_100Msketch_no0.98adj/di_mtrx\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "path = os.path.abspath(os.getcwd()) + \"/\"+ report_directory\n",
    "print(path)\n",
    "big_df.to_csv(os.path.join(path, r\"all_dist.csv\"), index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add coverage data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = os.path.abspath(os.getcwd()) + \"/\"+ report_directory + \"/\" + \"all_dist.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read df\n",
    "big_df = pd.read_csv(path, header = 0, sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>dist</th>\n",
       "      <th>cond</th>\n",
       "      <th>status</th>\n",
       "      <th>strap_iter</th>\n",
       "      <th>cond_tested</th>\n",
       "      <th>est_true</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>n2</td>\n",
       "      <td>n1</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>true_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>inf</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>n2</td>\n",
       "      <td>n1</td>\n",
       "      <td>0.249125</td>\n",
       "      <td>est_dist</td>\n",
       "      <td>1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>0.996500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>n2</td>\n",
       "      <td>n1</td>\n",
       "      <td>0.246457</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>0.985827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>n2</td>\n",
       "      <td>n1</td>\n",
       "      <td>0.247027</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>0.988106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>n2</td>\n",
       "      <td>n1</td>\n",
       "      <td>0.249732</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>1</td>\n",
       "      <td>10.0</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>0.998928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>n2</td>\n",
       "      <td>n1</td>\n",
       "      <td>0.247188</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>0.988753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>n2</td>\n",
       "      <td>n1</td>\n",
       "      <td>0.248988</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>1</td>\n",
       "      <td>12.0</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>0.995950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>n2</td>\n",
       "      <td>n1</td>\n",
       "      <td>0.247038</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>1</td>\n",
       "      <td>13.0</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>0.988151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>n2</td>\n",
       "      <td>n1</td>\n",
       "      <td>0.246600</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>1</td>\n",
       "      <td>14.0</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>0.986402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>n2</td>\n",
       "      <td>n1</td>\n",
       "      <td>0.244846</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>1</td>\n",
       "      <td>15.0</td>\n",
       "      <td>d0.25</td>\n",
       "      <td>0.979385</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample variable      dist       cond  status  strap_iter cond_tested  \\\n",
       "0     n2       n1  0.250000  true_dist       1         inf       d0.25   \n",
       "1     n2       n1  0.249125   est_dist       1        -1.0       d0.25   \n",
       "2     n2       n1  0.246457      d0.25       1         0.0       d0.25   \n",
       "3     n2       n1  0.247027      d0.25       1         1.0       d0.25   \n",
       "4     n2       n1  0.249732      d0.25       1        10.0       d0.25   \n",
       "5     n2       n1  0.247188      d0.25       1        11.0       d0.25   \n",
       "6     n2       n1  0.248988      d0.25       1        12.0       d0.25   \n",
       "7     n2       n1  0.247038      d0.25       1        13.0       d0.25   \n",
       "8     n2       n1  0.246600      d0.25       1        14.0       d0.25   \n",
       "9     n2       n1  0.244846      d0.25       1        15.0       d0.25   \n",
       "\n",
       "   est_true  \n",
       "0  1.000000  \n",
       "1  0.996500  \n",
       "2  0.985827  \n",
       "3  0.988106  \n",
       "4  0.998928  \n",
       "5  0.988753  \n",
       "6  0.995950  \n",
       "7  0.988151  \n",
       "8  0.986402  \n",
       "9  0.979385  "
      ]
     },
     "execution_count": 263,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "big_df.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read coverage information\n",
    "testsite_array = []\n",
    "with open(report_directory +'/coverage.txt') as my_file:\n",
    "    testsite_array = my_file.readlines()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['strap_0/library/n1_contigs.fa_0coverage\\t0.8024104680669222\\n', 'strap_0/library/n2_contigs.fa_0coverage\\t0.8727922300206834\\n', 'strap_10/library/n1_contigs.fa_10coverage\\t0.5099906126132728\\n', 'strap_10/library/n2_contigs.fa_10coverage\\t0.8976829685626065\\n', 'strap_11/library/n1_contigs.fa_11coverage\\t0.9326555012141516\\n', 'strap_11/library/n2_contigs.fa_11coverage\\t0.5479840972394283\\n', 'strap_12/library/n1_contigs.fa_12coverage\\t0.39962381211196574\\n', 'strap_12/library/n2_contigs.fa_12coverage\\t0.559495421115375\\n', 'strap_13/library/n1_contigs.fa_13coverage\\t0.3240259788064635\\n', 'strap_13/library/n2_contigs.fa_13coverage\\t0.7337739110633624\\n', 'strap_14/library/n1_contigs.fa_14coverage\\t0.7569504214946675\\n', 'strap_14/library/n2_contigs.fa_14coverage\\t0.7192575335842518\\n', 'strap_15/library/n1_contigs.fa_15coverage\\t0.5313291341782297\\n', 'strap_15/library/n2_contigs.fa_15coverage\\t0.8503795522082543\\n', 'strap_16/library/n1_contigs.fa_16coverage\\t0.21318499720745354\\n', 'strap_16/library/n2_contigs.fa_16coverage\\t0.6083446747080533\\n', 'strap_17/library/n1_contigs.fa_17coverage\\t0.7948083001605388\\n', 'strap_17/library/n2_contigs.fa_17coverage\\t0.7429949858032466\\n', 'strap_18/library/n1_contigs.fa_18coverage\\t0.33298149076734934\\n', 'strap_18/library/n2_contigs.fa_18coverage\\t0.6497052078419884\\n', 'strap_19/library/n1_contigs.fa_19coverage\\t0.5909132433859706\\n', 'strap_19/library/n2_contigs.fa_19coverage\\t0.23017812502142956\\n', 'strap_1/library/n1_contigs.fa_1coverage\\t0.34822040377015545\\n', 'strap_1/library/n2_contigs.fa_1coverage\\t0.908632914291612\\n', 'strap_20/library/n1_contigs.fa_20coverage\\t0.46795812503880524\\n', 'strap_20/library/n2_contigs.fa_20coverage\\t0.6575387959906736\\n', 'strap_21/library/n1_contigs.fa_21coverage\\t0.3287536107321826\\n', 'strap_21/library/n2_contigs.fa_21coverage\\t0.43775716710083823\\n', 'strap_22/library/n1_contigs.fa_22coverage\\t0.271459140006617\\n', 'strap_22/library/n2_contigs.fa_22coverage\\t0.5709714653662652\\n', 'strap_23/library/n1_contigs.fa_23coverage\\t0.776972071710823\\n', 'strap_23/library/n2_contigs.fa_23coverage\\t0.8531244131308738\\n', 'strap_24/library/n1_contigs.fa_24coverage\\t0.3384687990568043\\n', 'strap_24/library/n2_contigs.fa_24coverage\\t0.530180479033257\\n', 'strap_25/library/n1_contigs.fa_25coverage\\t0.2504295553143581\\n', 'strap_25/library/n2_contigs.fa_25coverage\\t0.20181333828416587\\n', 'strap_26/library/n1_contigs.fa_26coverage\\t0.6492799778553878\\n', 'strap_26/library/n2_contigs.fa_26coverage\\t0.5867484084707428\\n', 'strap_27/library/n1_contigs.fa_27coverage\\t0.24108473197506325\\n', 'strap_27/library/n2_contigs.fa_27coverage\\t0.8631850601530261\\n', 'strap_28/library/n1_contigs.fa_28coverage\\t0.6143893678978\\n', 'strap_28/library/n2_contigs.fa_28coverage\\t0.3497133487587611\\n', 'strap_29/library/n1_contigs.fa_29coverage\\t0.2724463200290049\\n', 'strap_29/library/n2_contigs.fa_29coverage\\t0.770155663778546\\n', 'strap_2/library/n1_contigs.fa_2coverage\\t0.5040269239302159\\n', 'strap_2/library/n2_contigs.fa_2coverage\\t0.7926163122751776\\n', 'strap_30/library/n1_contigs.fa_30coverage\\t0.9163390034842521\\n', 'strap_30/library/n2_contigs.fa_30coverage\\t0.3786017837421774\\n', 'strap_31/library/n1_contigs.fa_31coverage\\t0.21624519651977003\\n', 'strap_31/library/n2_contigs.fa_31coverage\\t0.2664223653623751\\n', 'strap_32/library/n1_contigs.fa_32coverage\\t0.8819920625563924\\n', 'strap_32/library/n2_contigs.fa_32coverage\\t0.5232841789735921\\n', 'strap_33/library/n1_contigs.fa_33coverage\\t0.540752837005317\\n', 'strap_33/library/n2_contigs.fa_33coverage\\t0.7222609076138462\\n', 'strap_34/library/n1_contigs.fa_34coverage\\t0.4298102761162747\\n', 'strap_34/library/n2_contigs.fa_34coverage\\t0.8895271053002064\\n', 'strap_35/library/n1_contigs.fa_35coverage\\t0.35930727958811826\\n', 'strap_35/library/n2_contigs.fa_35coverage\\t0.7026256223438268\\n', 'strap_36/library/n1_contigs.fa_36coverage\\t0.2956921108290727\\n', 'strap_36/library/n2_contigs.fa_36coverage\\t0.8744776371597195\\n', 'strap_37/library/n1_contigs.fa_37coverage\\t0.7817914831224668\\n', 'strap_37/library/n2_contigs.fa_37coverage\\t0.5620313218829153\\n', 'strap_38/library/n1_contigs.fa_38coverage\\t0.858202025764631\\n', 'strap_38/library/n2_contigs.fa_38coverage\\t0.5153224866355997\\n', 'strap_39/library/n1_contigs.fa_39coverage\\t0.7423900434460513\\n', 'strap_39/library/n2_contigs.fa_39coverage\\t0.19137261533631514\\n', 'strap_3/library/n1_contigs.fa_3coverage\\t0.8363299891739092\\n', 'strap_3/library/n2_contigs.fa_3coverage\\t0.3390497583438881\\n', 'strap_40/library/n1_contigs.fa_40coverage\\t0.9156850368369154\\n', 'strap_40/library/n2_contigs.fa_40coverage\\t0.6471748866025329\\n', 'strap_41/library/n1_contigs.fa_41coverage\\t0.4730495735332927\\n', 'strap_41/library/n2_contigs.fa_41coverage\\t0.7148945088849643\\n', 'strap_42/library/n1_contigs.fa_42coverage\\t0.23868959490663533\\n', 'strap_42/library/n2_contigs.fa_42coverage\\t0.318146441610965\\n', 'strap_43/library/n1_contigs.fa_43coverage\\t0.24471436944151423\\n', 'strap_43/library/n2_contigs.fa_43coverage\\t0.6413492712020048\\n', 'strap_44/library/n1_contigs.fa_44coverage\\t0.3491549538344498\\n', 'strap_44/library/n2_contigs.fa_44coverage\\t0.7631996241164644\\n', 'strap_45/library/n1_contigs.fa_45coverage\\t0.21471616341521121\\n', 'strap_45/library/n2_contigs.fa_45coverage\\t0.7867800141329733\\n', 'strap_46/library/n1_contigs.fa_46coverage\\t0.4644953049788333\\n', 'strap_46/library/n2_contigs.fa_46coverage\\t0.2463841908617301\\n', 'strap_47/library/n1_contigs.fa_47coverage\\t0.6883925877148983\\n', 'strap_47/library/n2_contigs.fa_47coverage\\t0.41302593844905156\\n', 'strap_48/library/n1_contigs.fa_48coverage\\t0.8497125700227077\\n', 'strap_48/library/n2_contigs.fa_48coverage\\t0.8762632811711732\\n', 'strap_49/library/n1_contigs.fa_49coverage\\t0.6119230591294005\\n', 'strap_49/library/n2_contigs.fa_49coverage\\t0.20366822535647236\\n', 'strap_4/library/n1_contigs.fa_4coverage\\t0.1977550660329987\\n', 'strap_4/library/n2_contigs.fa_4coverage\\t0.6908769477594289\\n', 'strap_50/library/n1_contigs.fa_50coverage\\t0.738826719889973\\n', 'strap_50/library/n2_contigs.fa_50coverage\\t0.458853660859337\\n', 'strap_51/library/n1_contigs.fa_51coverage\\t0.20900682495073025\\n', 'strap_51/library/n2_contigs.fa_51coverage\\t0.9241771766353223\\n', 'strap_52/library/n1_contigs.fa_52coverage\\t0.31882499888904714\\n', 'strap_52/library/n2_contigs.fa_52coverage\\t0.26150179845017874\\n', 'strap_53/library/n1_contigs.fa_53coverage\\t0.9492774170755022\\n', 'strap_53/library/n2_contigs.fa_53coverage\\t0.21507803095881556\\n', 'strap_54/library/n1_contigs.fa_54coverage\\t0.9546039084312843\\n', 'strap_54/library/n2_contigs.fa_54coverage\\t0.17951414007065505\\n', 'strap_55/library/n1_contigs.fa_55coverage\\t0.5244313657148529\\n', 'strap_55/library/n2_contigs.fa_55coverage\\t0.6937111031110633\\n', 'strap_56/library/n1_contigs.fa_56coverage\\t0.27773927400453247\\n', 'strap_56/library/n2_contigs.fa_56coverage\\t0.6896036898708257\\n', 'strap_57/library/n1_contigs.fa_57coverage\\t0.5635480666888711\\n', 'strap_57/library/n2_contigs.fa_57coverage\\t0.9391977625909539\\n', 'strap_58/library/n1_contigs.fa_58coverage\\t0.21833535374723095\\n', 'strap_58/library/n2_contigs.fa_58coverage\\t0.6827430714447025\\n', 'strap_59/library/n1_contigs.fa_59coverage\\t0.601659279509555\\n', 'strap_59/library/n2_contigs.fa_59coverage\\t0.8634431261166352\\n', 'strap_5/library/n1_contigs.fa_5coverage\\t0.3336947912256118\\n', 'strap_5/library/n2_contigs.fa_5coverage\\t0.6182578644922396\\n', 'strap_60/library/n1_contigs.fa_60coverage\\t0.8270019564055021\\n', 'strap_60/library/n2_contigs.fa_60coverage\\t0.3447163551301043\\n', 'strap_61/library/n1_contigs.fa_61coverage\\t0.5640863172620442\\n', 'strap_61/library/n2_contigs.fa_61coverage\\t0.9083028015650945\\n', 'strap_62/library/n1_contigs.fa_62coverage\\t0.7610702215308868\\n', 'strap_62/library/n2_contigs.fa_62coverage\\t0.4093613476150239\\n', 'strap_63/library/n1_contigs.fa_63coverage\\t0.6043413545251319\\n', 'strap_63/library/n2_contigs.fa_63coverage\\t0.7122476941708936\\n', 'strap_64/library/n1_contigs.fa_64coverage\\t0.49849073817802675\\n', 'strap_64/library/n2_contigs.fa_64coverage\\t0.7903607595582032\\n', 'strap_65/library/n1_contigs.fa_65coverage\\t0.5413069605565061\\n', 'strap_65/library/n2_contigs.fa_65coverage\\t0.7879544704762081\\n', 'strap_66/library/n1_contigs.fa_66coverage\\t0.6340141159899284\\n', 'strap_66/library/n2_contigs.fa_66coverage\\t0.8943172484681843\\n', 'strap_67/library/n1_contigs.fa_67coverage\\t0.664297219930901\\n', 'strap_67/library/n2_contigs.fa_67coverage\\t0.44124309458040684\\n', 'strap_68/library/n1_contigs.fa_68coverage\\t0.3811420454476135\\n', 'strap_68/library/n2_contigs.fa_68coverage\\t0.5918989324427366\\n', 'strap_69/library/n1_contigs.fa_69coverage\\t0.5005492773407182\\n', 'strap_69/library/n2_contigs.fa_69coverage\\t0.9159340509240508\\n', 'strap_6/library/n1_contigs.fa_6coverage\\t0.9445377311265982\\n', 'strap_6/library/n2_contigs.fa_6coverage\\t0.6842807444245481\\n', 'strap_70/library/n1_contigs.fa_70coverage\\t0.23277186611756023\\n', 'strap_70/library/n2_contigs.fa_70coverage\\t0.9312761044954533\\n', 'strap_71/library/n1_contigs.fa_71coverage\\t0.7601715385204723\\n', 'strap_71/library/n2_contigs.fa_71coverage\\t0.8579262112160623\\n', 'strap_72/library/n1_contigs.fa_72coverage\\t0.6461634058857245\\n', 'strap_72/library/n2_contigs.fa_72coverage\\t0.23069309068483684\\n', 'strap_73/library/n1_contigs.fa_73coverage\\t0.5267210161164859\\n', 'strap_73/library/n2_contigs.fa_73coverage\\t0.6009620932923209\\n', 'strap_74/library/n1_contigs.fa_74coverage\\t0.4867759294082198\\n', 'strap_74/library/n2_contigs.fa_74coverage\\t0.4356413455091568\\n', 'strap_75/library/n1_contigs.fa_75coverage\\t0.7232151362662067\\n', 'strap_75/library/n2_contigs.fa_75coverage\\t0.4170329534472561\\n', 'strap_76/library/n1_contigs.fa_76coverage\\t0.7443784931327401\\n', 'strap_76/library/n2_contigs.fa_76coverage\\t0.888841978564651\\n', 'strap_77/library/n1_contigs.fa_77coverage\\t0.5285642588369976\\n', 'strap_77/library/n2_contigs.fa_77coverage\\t0.47218734056771505\\n', 'strap_78/library/n1_contigs.fa_78coverage\\t0.6874330780864333\\n', 'strap_78/library/n2_contigs.fa_78coverage\\t0.37214072494152756\\n', 'strap_79/library/n1_contigs.fa_79coverage\\t0.9327942850309208\\n', 'strap_79/library/n2_contigs.fa_79coverage\\t0.32565067416871885\\n', 'strap_7/library/n1_contigs.fa_7coverage\\t0.46238406182665576\\n', 'strap_7/library/n2_contigs.fa_7coverage\\t0.611447112734372\\n', 'strap_80/library/n1_contigs.fa_80coverage\\t0.20705205380017405\\n', 'strap_80/library/n2_contigs.fa_80coverage\\t0.8245243554124112\\n', 'strap_81/library/n1_contigs.fa_81coverage\\t0.7570778249773294\\n', 'strap_81/library/n2_contigs.fa_81coverage\\t0.5150548940807326\\n', 'strap_82/library/n1_contigs.fa_82coverage\\t0.6259786029322579\\n', 'strap_82/library/n2_contigs.fa_82coverage\\t0.8075845162302935\\n', 'strap_83/library/n1_contigs.fa_83coverage\\t0.6589611920669614\\n', 'strap_83/library/n2_contigs.fa_83coverage\\t0.8168147913339784\\n', 'strap_84/library/n1_contigs.fa_84coverage\\t0.7771088128842564\\n', 'strap_84/library/n2_contigs.fa_84coverage\\t0.24033844732736748\\n', 'strap_85/library/n1_contigs.fa_85coverage\\t0.2786541185851564\\n', 'strap_85/library/n2_contigs.fa_85coverage\\t0.4746460547654805\\n', 'strap_86/library/n1_contigs.fa_86coverage\\t0.3262083845366707\\n', 'strap_86/library/n2_contigs.fa_86coverage\\t0.45937256815511013\\n', 'strap_87/library/n1_contigs.fa_87coverage\\t0.5821028716216052\\n', 'strap_87/library/n2_contigs.fa_87coverage\\t0.4571272844250626\\n', 'strap_88/library/n1_contigs.fa_88coverage\\t0.4873158868431143\\n', 'strap_88/library/n2_contigs.fa_88coverage\\t0.775382608506199\\n', 'strap_89/library/n1_contigs.fa_89coverage\\t0.3183910806248716\\n', 'strap_89/library/n2_contigs.fa_89coverage\\t0.7628997599619655\\n', 'strap_8/library/n1_contigs.fa_8coverage\\t0.6492781030214154\\n', 'strap_8/library/n2_contigs.fa_8coverage\\t0.32707489992557537\\n', 'strap_90/library/n1_contigs.fa_90coverage\\t0.7113023898023962\\n', 'strap_90/library/n2_contigs.fa_90coverage\\t0.20829577872961155\\n', 'strap_91/library/n1_contigs.fa_91coverage\\t0.2985291147615028\\n', 'strap_91/library/n2_contigs.fa_91coverage\\t0.7905159417741708\\n', 'strap_92/library/n1_contigs.fa_92coverage\\t0.29298850748889477\\n', 'strap_92/library/n2_contigs.fa_92coverage\\t0.7065074134775788\\n', 'strap_93/library/n1_contigs.fa_93coverage\\t0.7706095026390161\\n', 'strap_93/library/n2_contigs.fa_93coverage\\t0.19106480599692444\\n', 'strap_94/library/n1_contigs.fa_94coverage\\t0.8577825555373964\\n', 'strap_94/library/n2_contigs.fa_94coverage\\t0.22135329182232222\\n', 'strap_95/library/n1_contigs.fa_95coverage\\t0.4090168034349715\\n', 'strap_95/library/n2_contigs.fa_95coverage\\t0.7854468586564988\\n', 'strap_96/library/n1_contigs.fa_96coverage\\t0.5601099066112899\\n', 'strap_96/library/n2_contigs.fa_96coverage\\t0.8862921326369054\\n', 'strap_97/library/n1_contigs.fa_97coverage\\t0.6442335535895731\\n', 'strap_97/library/n2_contigs.fa_97coverage\\t0.22883947325084728\\n', 'strap_98/library/n1_contigs.fa_98coverage\\t0.8484329274515531\\n', 'strap_98/library/n2_contigs.fa_98coverage\\t0.24914889062365703\\n', 'strap_99/library/n1_contigs.fa_99coverage\\t0.337315214430537\\n', 'strap_99/library/n2_contigs.fa_99coverage\\t0.3180509778590529\\n', 'strap_9/library/n1_contigs.fa_9coverage\\t0.9203366035153175\\n', 'strap_9/library/n2_contigs.fa_9coverage\\t0.8934205824637712\\n']\n"
     ]
    }
   ],
   "source": [
    "print(testsite_array)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "big_df[\"cov_smpl\"] = ''\n",
    "big_df[\"cov_var\"] = ''\n",
    "big_df[\"cov_tag\"] = -1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "def set_cov_tag(row):\n",
    "    ret = float('inf')\n",
    "    \n",
    "    if row['cov_smpl'] == \"NA\" and row['cov_var'] == \"NA\":\n",
    "        ret = 2\n",
    "        \n",
    "    elif row['cov_smpl'] == \"NA\" or row['cov_var'] == \"NA\":\n",
    "        ret = 1\n",
    "        \n",
    "    elif row['cov_smpl'] == \"\" and row['cov_var'] == \"\":\n",
    "        ret =  float('inf')\n",
    "        \n",
    "    else:\n",
    "        ret =  0\n",
    "        \n",
    "    return ret"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['strap_0/library/n1_contigs.fa_0coverage\\t0.8024104680669222\\n']\n",
      "0.8024104680669222\n",
      "['strap_0/library/n2_contigs.fa_0coverage\\t0.8727922300206834\\n']\n",
      "0.8727922300206834\n",
      "['strap_1/library/n1_contigs.fa_1coverage\\t0.34822040377015545\\n']\n",
      "0.34822040377015545\n",
      "['strap_1/library/n2_contigs.fa_1coverage\\t0.908632914291612\\n']\n",
      "0.908632914291612\n",
      "['strap_10/library/n1_contigs.fa_10coverage\\t0.5099906126132728\\n']\n",
      "0.5099906126132728\n",
      "['strap_10/library/n2_contigs.fa_10coverage\\t0.8976829685626065\\n']\n",
      "0.8976829685626065\n",
      "['strap_11/library/n1_contigs.fa_11coverage\\t0.9326555012141516\\n']\n",
      "0.9326555012141516\n",
      "['strap_11/library/n2_contigs.fa_11coverage\\t0.5479840972394283\\n']\n",
      "0.5479840972394283\n",
      "['strap_12/library/n1_contigs.fa_12coverage\\t0.39962381211196574\\n']\n",
      "0.39962381211196574\n",
      "['strap_12/library/n2_contigs.fa_12coverage\\t0.559495421115375\\n']\n",
      "0.559495421115375\n",
      "['strap_13/library/n1_contigs.fa_13coverage\\t0.3240259788064635\\n']\n",
      "0.3240259788064635\n",
      "['strap_13/library/n2_contigs.fa_13coverage\\t0.7337739110633624\\n']\n",
      "0.7337739110633624\n",
      "['strap_14/library/n1_contigs.fa_14coverage\\t0.7569504214946675\\n']\n",
      "0.7569504214946675\n",
      "['strap_14/library/n2_contigs.fa_14coverage\\t0.7192575335842518\\n']\n",
      "0.7192575335842518\n",
      "['strap_15/library/n1_contigs.fa_15coverage\\t0.5313291341782297\\n']\n",
      "0.5313291341782297\n",
      "['strap_15/library/n2_contigs.fa_15coverage\\t0.8503795522082543\\n']\n",
      "0.8503795522082543\n",
      "['strap_16/library/n1_contigs.fa_16coverage\\t0.21318499720745354\\n']\n",
      "0.21318499720745354\n",
      "['strap_16/library/n2_contigs.fa_16coverage\\t0.6083446747080533\\n']\n",
      "0.6083446747080533\n",
      "['strap_17/library/n1_contigs.fa_17coverage\\t0.7948083001605388\\n']\n",
      "0.7948083001605388\n",
      "['strap_17/library/n2_contigs.fa_17coverage\\t0.7429949858032466\\n']\n",
      "0.7429949858032466\n",
      "['strap_18/library/n1_contigs.fa_18coverage\\t0.33298149076734934\\n']\n",
      "0.33298149076734934\n",
      "['strap_18/library/n2_contigs.fa_18coverage\\t0.6497052078419884\\n']\n",
      "0.6497052078419884\n",
      "['strap_19/library/n1_contigs.fa_19coverage\\t0.5909132433859706\\n']\n",
      "0.5909132433859706\n",
      "['strap_19/library/n2_contigs.fa_19coverage\\t0.23017812502142956\\n']\n",
      "0.23017812502142956\n",
      "['strap_2/library/n1_contigs.fa_2coverage\\t0.5040269239302159\\n']\n",
      "0.5040269239302159\n",
      "['strap_2/library/n2_contigs.fa_2coverage\\t0.7926163122751776\\n']\n",
      "0.7926163122751776\n",
      "['strap_20/library/n1_contigs.fa_20coverage\\t0.46795812503880524\\n']\n",
      "0.46795812503880524\n",
      "['strap_20/library/n2_contigs.fa_20coverage\\t0.6575387959906736\\n']\n",
      "0.6575387959906736\n",
      "['strap_21/library/n1_contigs.fa_21coverage\\t0.3287536107321826\\n']\n",
      "0.3287536107321826\n",
      "['strap_21/library/n2_contigs.fa_21coverage\\t0.43775716710083823\\n']\n",
      "0.43775716710083823\n",
      "['strap_22/library/n1_contigs.fa_22coverage\\t0.271459140006617\\n']\n",
      "0.271459140006617\n",
      "['strap_22/library/n2_contigs.fa_22coverage\\t0.5709714653662652\\n']\n",
      "0.5709714653662652\n",
      "['strap_23/library/n1_contigs.fa_23coverage\\t0.776972071710823\\n']\n",
      "0.776972071710823\n",
      "['strap_23/library/n2_contigs.fa_23coverage\\t0.8531244131308738\\n']\n",
      "0.8531244131308738\n",
      "['strap_24/library/n1_contigs.fa_24coverage\\t0.3384687990568043\\n']\n",
      "0.3384687990568043\n",
      "['strap_24/library/n2_contigs.fa_24coverage\\t0.530180479033257\\n']\n",
      "0.530180479033257\n",
      "['strap_25/library/n1_contigs.fa_25coverage\\t0.2504295553143581\\n']\n",
      "0.2504295553143581\n",
      "['strap_25/library/n2_contigs.fa_25coverage\\t0.20181333828416587\\n']\n",
      "0.20181333828416587\n",
      "['strap_26/library/n1_contigs.fa_26coverage\\t0.6492799778553878\\n']\n",
      "0.6492799778553878\n",
      "['strap_26/library/n2_contigs.fa_26coverage\\t0.5867484084707428\\n']\n",
      "0.5867484084707428\n",
      "['strap_27/library/n1_contigs.fa_27coverage\\t0.24108473197506325\\n']\n",
      "0.24108473197506325\n",
      "['strap_27/library/n2_contigs.fa_27coverage\\t0.8631850601530261\\n']\n",
      "0.8631850601530261\n",
      "['strap_28/library/n1_contigs.fa_28coverage\\t0.6143893678978\\n']\n",
      "0.6143893678978\n",
      "['strap_28/library/n2_contigs.fa_28coverage\\t0.3497133487587611\\n']\n",
      "0.3497133487587611\n",
      "['strap_29/library/n1_contigs.fa_29coverage\\t0.2724463200290049\\n']\n",
      "0.2724463200290049\n",
      "['strap_29/library/n2_contigs.fa_29coverage\\t0.770155663778546\\n']\n",
      "0.770155663778546\n",
      "['strap_3/library/n1_contigs.fa_3coverage\\t0.8363299891739092\\n']\n",
      "0.8363299891739092\n",
      "['strap_3/library/n2_contigs.fa_3coverage\\t0.3390497583438881\\n']\n",
      "0.3390497583438881\n",
      "['strap_30/library/n1_contigs.fa_30coverage\\t0.9163390034842521\\n']\n",
      "0.9163390034842521\n",
      "['strap_30/library/n2_contigs.fa_30coverage\\t0.3786017837421774\\n']\n",
      "0.3786017837421774\n",
      "['strap_31/library/n1_contigs.fa_31coverage\\t0.21624519651977003\\n']\n",
      "0.21624519651977003\n",
      "['strap_31/library/n2_contigs.fa_31coverage\\t0.2664223653623751\\n']\n",
      "0.2664223653623751\n",
      "['strap_32/library/n1_contigs.fa_32coverage\\t0.8819920625563924\\n']\n",
      "0.8819920625563924\n",
      "['strap_32/library/n2_contigs.fa_32coverage\\t0.5232841789735921\\n']\n",
      "0.5232841789735921\n",
      "['strap_33/library/n1_contigs.fa_33coverage\\t0.540752837005317\\n']\n",
      "0.540752837005317\n",
      "['strap_33/library/n2_contigs.fa_33coverage\\t0.7222609076138462\\n']\n",
      "0.7222609076138462\n",
      "['strap_34/library/n1_contigs.fa_34coverage\\t0.4298102761162747\\n']\n",
      "0.4298102761162747\n",
      "['strap_34/library/n2_contigs.fa_34coverage\\t0.8895271053002064\\n']\n",
      "0.8895271053002064\n",
      "['strap_35/library/n1_contigs.fa_35coverage\\t0.35930727958811826\\n']\n",
      "0.35930727958811826\n",
      "['strap_35/library/n2_contigs.fa_35coverage\\t0.7026256223438268\\n']\n",
      "0.7026256223438268\n",
      "['strap_36/library/n1_contigs.fa_36coverage\\t0.2956921108290727\\n']\n",
      "0.2956921108290727\n",
      "['strap_36/library/n2_contigs.fa_36coverage\\t0.8744776371597195\\n']\n",
      "0.8744776371597195\n",
      "['strap_37/library/n1_contigs.fa_37coverage\\t0.7817914831224668\\n']\n",
      "0.7817914831224668\n",
      "['strap_37/library/n2_contigs.fa_37coverage\\t0.5620313218829153\\n']\n",
      "0.5620313218829153\n",
      "['strap_38/library/n1_contigs.fa_38coverage\\t0.858202025764631\\n']\n",
      "0.858202025764631\n",
      "['strap_38/library/n2_contigs.fa_38coverage\\t0.5153224866355997\\n']\n",
      "0.5153224866355997\n",
      "['strap_39/library/n1_contigs.fa_39coverage\\t0.7423900434460513\\n']\n",
      "0.7423900434460513\n",
      "['strap_39/library/n2_contigs.fa_39coverage\\t0.19137261533631514\\n']\n",
      "0.19137261533631514\n",
      "['strap_4/library/n1_contigs.fa_4coverage\\t0.1977550660329987\\n']\n",
      "0.1977550660329987\n",
      "['strap_4/library/n2_contigs.fa_4coverage\\t0.6908769477594289\\n']\n",
      "0.6908769477594289\n",
      "['strap_40/library/n1_contigs.fa_40coverage\\t0.9156850368369154\\n']\n",
      "0.9156850368369154\n",
      "['strap_40/library/n2_contigs.fa_40coverage\\t0.6471748866025329\\n']\n",
      "0.6471748866025329\n",
      "['strap_41/library/n1_contigs.fa_41coverage\\t0.4730495735332927\\n']\n",
      "0.4730495735332927\n",
      "['strap_41/library/n2_contigs.fa_41coverage\\t0.7148945088849643\\n']\n",
      "0.7148945088849643\n",
      "['strap_42/library/n1_contigs.fa_42coverage\\t0.23868959490663533\\n']\n",
      "0.23868959490663533\n",
      "['strap_42/library/n2_contigs.fa_42coverage\\t0.318146441610965\\n']\n",
      "0.318146441610965\n",
      "['strap_43/library/n1_contigs.fa_43coverage\\t0.24471436944151423\\n']\n",
      "0.24471436944151423\n",
      "['strap_43/library/n2_contigs.fa_43coverage\\t0.6413492712020048\\n']\n",
      "0.6413492712020048\n",
      "['strap_44/library/n1_contigs.fa_44coverage\\t0.3491549538344498\\n']\n",
      "0.3491549538344498\n",
      "['strap_44/library/n2_contigs.fa_44coverage\\t0.7631996241164644\\n']\n",
      "0.7631996241164644\n",
      "['strap_45/library/n1_contigs.fa_45coverage\\t0.21471616341521121\\n']\n",
      "0.21471616341521121\n",
      "['strap_45/library/n2_contigs.fa_45coverage\\t0.7867800141329733\\n']\n",
      "0.7867800141329733\n",
      "['strap_46/library/n1_contigs.fa_46coverage\\t0.4644953049788333\\n']\n",
      "0.4644953049788333\n",
      "['strap_46/library/n2_contigs.fa_46coverage\\t0.2463841908617301\\n']\n",
      "0.2463841908617301\n",
      "['strap_47/library/n1_contigs.fa_47coverage\\t0.6883925877148983\\n']\n",
      "0.6883925877148983\n",
      "['strap_47/library/n2_contigs.fa_47coverage\\t0.41302593844905156\\n']\n",
      "0.41302593844905156\n",
      "['strap_48/library/n1_contigs.fa_48coverage\\t0.8497125700227077\\n']\n",
      "0.8497125700227077\n",
      "['strap_48/library/n2_contigs.fa_48coverage\\t0.8762632811711732\\n']\n",
      "0.8762632811711732\n",
      "['strap_49/library/n1_contigs.fa_49coverage\\t0.6119230591294005\\n']\n",
      "0.6119230591294005\n",
      "['strap_49/library/n2_contigs.fa_49coverage\\t0.20366822535647236\\n']\n",
      "0.20366822535647236\n",
      "['strap_5/library/n1_contigs.fa_5coverage\\t0.3336947912256118\\n']\n",
      "0.3336947912256118\n",
      "['strap_5/library/n2_contigs.fa_5coverage\\t0.6182578644922396\\n']\n",
      "0.6182578644922396\n",
      "['strap_50/library/n1_contigs.fa_50coverage\\t0.738826719889973\\n']\n",
      "0.738826719889973\n",
      "['strap_50/library/n2_contigs.fa_50coverage\\t0.458853660859337\\n']\n",
      "0.458853660859337\n",
      "['strap_51/library/n1_contigs.fa_51coverage\\t0.20900682495073025\\n']\n",
      "0.20900682495073025\n",
      "['strap_51/library/n2_contigs.fa_51coverage\\t0.9241771766353223\\n']\n",
      "0.9241771766353223\n",
      "['strap_52/library/n1_contigs.fa_52coverage\\t0.31882499888904714\\n']\n",
      "0.31882499888904714\n",
      "['strap_52/library/n2_contigs.fa_52coverage\\t0.26150179845017874\\n']\n",
      "0.26150179845017874\n",
      "['strap_53/library/n1_contigs.fa_53coverage\\t0.9492774170755022\\n']\n",
      "0.9492774170755022\n",
      "['strap_53/library/n2_contigs.fa_53coverage\\t0.21507803095881556\\n']\n",
      "0.21507803095881556\n",
      "['strap_54/library/n1_contigs.fa_54coverage\\t0.9546039084312843\\n']\n",
      "0.9546039084312843\n",
      "['strap_54/library/n2_contigs.fa_54coverage\\t0.17951414007065505\\n']\n",
      "0.17951414007065505\n",
      "['strap_55/library/n1_contigs.fa_55coverage\\t0.5244313657148529\\n']\n",
      "0.5244313657148529\n",
      "['strap_55/library/n2_contigs.fa_55coverage\\t0.6937111031110633\\n']\n",
      "0.6937111031110633\n",
      "['strap_56/library/n1_contigs.fa_56coverage\\t0.27773927400453247\\n']\n",
      "0.27773927400453247\n",
      "['strap_56/library/n2_contigs.fa_56coverage\\t0.6896036898708257\\n']\n",
      "0.6896036898708257\n",
      "['strap_57/library/n1_contigs.fa_57coverage\\t0.5635480666888711\\n']\n",
      "0.5635480666888711\n",
      "['strap_57/library/n2_contigs.fa_57coverage\\t0.9391977625909539\\n']\n",
      "0.9391977625909539\n",
      "['strap_58/library/n1_contigs.fa_58coverage\\t0.21833535374723095\\n']\n",
      "0.21833535374723095\n",
      "['strap_58/library/n2_contigs.fa_58coverage\\t0.6827430714447025\\n']\n",
      "0.6827430714447025\n",
      "['strap_59/library/n1_contigs.fa_59coverage\\t0.601659279509555\\n']\n",
      "0.601659279509555\n",
      "['strap_59/library/n2_contigs.fa_59coverage\\t0.8634431261166352\\n']\n",
      "0.8634431261166352\n",
      "['strap_6/library/n1_contigs.fa_6coverage\\t0.9445377311265982\\n']\n",
      "0.9445377311265982\n",
      "['strap_6/library/n2_contigs.fa_6coverage\\t0.6842807444245481\\n']\n",
      "0.6842807444245481\n",
      "['strap_60/library/n1_contigs.fa_60coverage\\t0.8270019564055021\\n']\n",
      "0.8270019564055021\n",
      "['strap_60/library/n2_contigs.fa_60coverage\\t0.3447163551301043\\n']\n",
      "0.3447163551301043\n",
      "['strap_61/library/n1_contigs.fa_61coverage\\t0.5640863172620442\\n']\n",
      "0.5640863172620442\n",
      "['strap_61/library/n2_contigs.fa_61coverage\\t0.9083028015650945\\n']\n",
      "0.9083028015650945\n",
      "['strap_62/library/n1_contigs.fa_62coverage\\t0.7610702215308868\\n']\n",
      "0.7610702215308868\n",
      "['strap_62/library/n2_contigs.fa_62coverage\\t0.4093613476150239\\n']\n",
      "0.4093613476150239\n",
      "['strap_63/library/n1_contigs.fa_63coverage\\t0.6043413545251319\\n']\n",
      "0.6043413545251319\n",
      "['strap_63/library/n2_contigs.fa_63coverage\\t0.7122476941708936\\n']\n",
      "0.7122476941708936\n",
      "['strap_64/library/n1_contigs.fa_64coverage\\t0.49849073817802675\\n']\n",
      "0.49849073817802675\n",
      "['strap_64/library/n2_contigs.fa_64coverage\\t0.7903607595582032\\n']\n",
      "0.7903607595582032\n",
      "['strap_65/library/n1_contigs.fa_65coverage\\t0.5413069605565061\\n']\n",
      "0.5413069605565061\n",
      "['strap_65/library/n2_contigs.fa_65coverage\\t0.7879544704762081\\n']\n",
      "0.7879544704762081\n",
      "['strap_66/library/n1_contigs.fa_66coverage\\t0.6340141159899284\\n']\n",
      "0.6340141159899284\n",
      "['strap_66/library/n2_contigs.fa_66coverage\\t0.8943172484681843\\n']\n",
      "0.8943172484681843\n",
      "['strap_67/library/n1_contigs.fa_67coverage\\t0.664297219930901\\n']\n",
      "0.664297219930901\n",
      "['strap_67/library/n2_contigs.fa_67coverage\\t0.44124309458040684\\n']\n",
      "0.44124309458040684\n",
      "['strap_68/library/n1_contigs.fa_68coverage\\t0.3811420454476135\\n']\n",
      "0.3811420454476135\n",
      "['strap_68/library/n2_contigs.fa_68coverage\\t0.5918989324427366\\n']\n",
      "0.5918989324427366\n",
      "['strap_69/library/n1_contigs.fa_69coverage\\t0.5005492773407182\\n']\n",
      "0.5005492773407182\n",
      "['strap_69/library/n2_contigs.fa_69coverage\\t0.9159340509240508\\n']\n",
      "0.9159340509240508\n",
      "['strap_7/library/n1_contigs.fa_7coverage\\t0.46238406182665576\\n']\n",
      "0.46238406182665576\n",
      "['strap_7/library/n2_contigs.fa_7coverage\\t0.611447112734372\\n']\n",
      "0.611447112734372\n",
      "['strap_70/library/n1_contigs.fa_70coverage\\t0.23277186611756023\\n']\n",
      "0.23277186611756023\n",
      "['strap_70/library/n2_contigs.fa_70coverage\\t0.9312761044954533\\n']\n",
      "0.9312761044954533\n",
      "['strap_71/library/n1_contigs.fa_71coverage\\t0.7601715385204723\\n']\n",
      "0.7601715385204723\n",
      "['strap_71/library/n2_contigs.fa_71coverage\\t0.8579262112160623\\n']\n",
      "0.8579262112160623\n",
      "['strap_72/library/n1_contigs.fa_72coverage\\t0.6461634058857245\\n']\n",
      "0.6461634058857245\n",
      "['strap_72/library/n2_contigs.fa_72coverage\\t0.23069309068483684\\n']\n",
      "0.23069309068483684\n",
      "['strap_73/library/n1_contigs.fa_73coverage\\t0.5267210161164859\\n']\n",
      "0.5267210161164859\n",
      "['strap_73/library/n2_contigs.fa_73coverage\\t0.6009620932923209\\n']\n",
      "0.6009620932923209\n",
      "['strap_74/library/n1_contigs.fa_74coverage\\t0.4867759294082198\\n']\n",
      "0.4867759294082198\n",
      "['strap_74/library/n2_contigs.fa_74coverage\\t0.4356413455091568\\n']\n",
      "0.4356413455091568\n",
      "['strap_75/library/n1_contigs.fa_75coverage\\t0.7232151362662067\\n']\n",
      "0.7232151362662067\n",
      "['strap_75/library/n2_contigs.fa_75coverage\\t0.4170329534472561\\n']\n",
      "0.4170329534472561\n",
      "['strap_76/library/n1_contigs.fa_76coverage\\t0.7443784931327401\\n']\n",
      "0.7443784931327401\n",
      "['strap_76/library/n2_contigs.fa_76coverage\\t0.888841978564651\\n']\n",
      "0.888841978564651\n",
      "['strap_77/library/n1_contigs.fa_77coverage\\t0.5285642588369976\\n']\n",
      "0.5285642588369976\n",
      "['strap_77/library/n2_contigs.fa_77coverage\\t0.47218734056771505\\n']\n",
      "0.47218734056771505\n",
      "['strap_78/library/n1_contigs.fa_78coverage\\t0.6874330780864333\\n']\n",
      "0.6874330780864333\n",
      "['strap_78/library/n2_contigs.fa_78coverage\\t0.37214072494152756\\n']\n",
      "0.37214072494152756\n",
      "['strap_79/library/n1_contigs.fa_79coverage\\t0.9327942850309208\\n']\n",
      "0.9327942850309208\n",
      "['strap_79/library/n2_contigs.fa_79coverage\\t0.32565067416871885\\n']\n",
      "0.32565067416871885\n",
      "['strap_8/library/n1_contigs.fa_8coverage\\t0.6492781030214154\\n']\n",
      "0.6492781030214154\n",
      "['strap_8/library/n2_contigs.fa_8coverage\\t0.32707489992557537\\n']\n",
      "0.32707489992557537\n",
      "['strap_80/library/n1_contigs.fa_80coverage\\t0.20705205380017405\\n']\n",
      "0.20705205380017405\n",
      "['strap_80/library/n2_contigs.fa_80coverage\\t0.8245243554124112\\n']\n",
      "0.8245243554124112\n",
      "['strap_81/library/n1_contigs.fa_81coverage\\t0.7570778249773294\\n']\n",
      "0.7570778249773294\n",
      "['strap_81/library/n2_contigs.fa_81coverage\\t0.5150548940807326\\n']\n",
      "0.5150548940807326\n",
      "['strap_82/library/n1_contigs.fa_82coverage\\t0.6259786029322579\\n']\n",
      "0.6259786029322579\n",
      "['strap_82/library/n2_contigs.fa_82coverage\\t0.8075845162302935\\n']\n",
      "0.8075845162302935\n",
      "['strap_83/library/n1_contigs.fa_83coverage\\t0.6589611920669614\\n']\n",
      "0.6589611920669614\n",
      "['strap_83/library/n2_contigs.fa_83coverage\\t0.8168147913339784\\n']\n",
      "0.8168147913339784\n",
      "['strap_84/library/n1_contigs.fa_84coverage\\t0.7771088128842564\\n']\n",
      "0.7771088128842564\n",
      "['strap_84/library/n2_contigs.fa_84coverage\\t0.24033844732736748\\n']\n",
      "0.24033844732736748\n",
      "['strap_85/library/n1_contigs.fa_85coverage\\t0.2786541185851564\\n']\n",
      "0.2786541185851564\n",
      "['strap_85/library/n2_contigs.fa_85coverage\\t0.4746460547654805\\n']\n",
      "0.4746460547654805\n",
      "['strap_86/library/n1_contigs.fa_86coverage\\t0.3262083845366707\\n']\n",
      "0.3262083845366707\n",
      "['strap_86/library/n2_contigs.fa_86coverage\\t0.45937256815511013\\n']\n",
      "0.45937256815511013\n",
      "['strap_87/library/n1_contigs.fa_87coverage\\t0.5821028716216052\\n']\n",
      "0.5821028716216052\n",
      "['strap_87/library/n2_contigs.fa_87coverage\\t0.4571272844250626\\n']\n",
      "0.4571272844250626\n",
      "['strap_88/library/n1_contigs.fa_88coverage\\t0.4873158868431143\\n']\n",
      "0.4873158868431143\n",
      "['strap_88/library/n2_contigs.fa_88coverage\\t0.775382608506199\\n']\n",
      "0.775382608506199\n",
      "['strap_89/library/n1_contigs.fa_89coverage\\t0.3183910806248716\\n']\n",
      "0.3183910806248716\n",
      "['strap_89/library/n2_contigs.fa_89coverage\\t0.7628997599619655\\n']\n",
      "0.7628997599619655\n",
      "['strap_9/library/n1_contigs.fa_9coverage\\t0.9203366035153175\\n']\n",
      "0.9203366035153175\n",
      "['strap_9/library/n2_contigs.fa_9coverage\\t0.8934205824637712\\n']\n",
      "0.8934205824637712\n",
      "['strap_90/library/n1_contigs.fa_90coverage\\t0.7113023898023962\\n']\n",
      "0.7113023898023962\n",
      "['strap_90/library/n2_contigs.fa_90coverage\\t0.20829577872961155\\n']\n",
      "0.20829577872961155\n",
      "['strap_91/library/n1_contigs.fa_91coverage\\t0.2985291147615028\\n']\n",
      "0.2985291147615028\n",
      "['strap_91/library/n2_contigs.fa_91coverage\\t0.7905159417741708\\n']\n",
      "0.7905159417741708\n",
      "['strap_92/library/n1_contigs.fa_92coverage\\t0.29298850748889477\\n']\n",
      "0.29298850748889477\n",
      "['strap_92/library/n2_contigs.fa_92coverage\\t0.7065074134775788\\n']\n",
      "0.7065074134775788\n",
      "['strap_93/library/n1_contigs.fa_93coverage\\t0.7706095026390161\\n']\n",
      "0.7706095026390161\n",
      "['strap_93/library/n2_contigs.fa_93coverage\\t0.19106480599692444\\n']\n",
      "0.19106480599692444\n",
      "['strap_94/library/n1_contigs.fa_94coverage\\t0.8577825555373964\\n']\n",
      "0.8577825555373964\n",
      "['strap_94/library/n2_contigs.fa_94coverage\\t0.22135329182232222\\n']\n",
      "0.22135329182232222\n",
      "['strap_95/library/n1_contigs.fa_95coverage\\t0.4090168034349715\\n']\n",
      "0.4090168034349715\n",
      "['strap_95/library/n2_contigs.fa_95coverage\\t0.7854468586564988\\n']\n",
      "0.7854468586564988\n",
      "['strap_96/library/n1_contigs.fa_96coverage\\t0.5601099066112899\\n']\n",
      "0.5601099066112899\n",
      "['strap_96/library/n2_contigs.fa_96coverage\\t0.8862921326369054\\n']\n",
      "0.8862921326369054\n",
      "['strap_97/library/n1_contigs.fa_97coverage\\t0.6442335535895731\\n']\n",
      "0.6442335535895731\n",
      "['strap_97/library/n2_contigs.fa_97coverage\\t0.22883947325084728\\n']\n",
      "0.22883947325084728\n",
      "['strap_98/library/n1_contigs.fa_98coverage\\t0.8484329274515531\\n']\n",
      "0.8484329274515531\n",
      "['strap_98/library/n2_contigs.fa_98coverage\\t0.24914889062365703\\n']\n",
      "0.24914889062365703\n",
      "['strap_99/library/n1_contigs.fa_99coverage\\t0.337315214430537\\n']\n",
      "0.337315214430537\n",
      "['strap_99/library/n2_contigs.fa_99coverage\\t0.3180509778590529\\n']\n",
      "0.3180509778590529\n"
     ]
    }
   ],
   "source": [
    "for l in range (0, len(big_df)):\n",
    "    try:\n",
    "        strap_num = str(int(big_df.at[l, 'strap_iter']))\n",
    "    except:\n",
    "         strap_num = str(big_df.at[l, 'strap_iter'])\n",
    "            \n",
    "    \n",
    "    line_tag = \"strap_\" + strap_num + \"/library/\" + big_df.at[l, \"sample\"] + \"_contigs.fa_\" + strap_num + \"coverage\"\n",
    "    #line_tag = \"strap_\" + strap_num + \"/library/\" + big_df.at[l, \"sample\"] + \"_contigs_1xef.fa_\" + strap_num + \"coverage\"\n",
    "    #print(line_tag)\n",
    "    c = [i for i in testsite_array if line_tag in i ]\n",
    "    if len(c) > 0:\n",
    "        print(c)\n",
    "        t1 = re.split('[\\t\\n]', c[0])[-2]\n",
    "        print(t1)\n",
    "        big_df.at[l, \"cov_smpl\"] = t1\n",
    "    \n",
    "    \n",
    "    line_tag = \"strap_\" + strap_num + \"/library/\" + big_df.at[l, \"variable\"] + \"_contigs.fa_\" + strap_num + \"coverage\"\n",
    "    #line_tag = \"strap_\" + strap_num + \"/library/\" + big_df.at[l, \"variable\"] + \"_contigs_1xef.fa_\" + strap_num + \"coverage\"\n",
    "    #print(line_tag)\n",
    "    d = [i for i in testsite_array if line_tag in i ]\n",
    "    if len(d) > 0:\n",
    "        print(d)\n",
    "        t2 = re.split('[\\t\\n]', d[0])[-2]\n",
    "        print(t2)\n",
    "        big_df.at[l, \"cov_var\"] = t2\n",
    "    \n",
    "    \n",
    "    \n",
    "\n",
    "big_df['cov_tag'] =  big_df.apply(set_cov_tag, axis=1)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/admin/Documents/support/zombi_cherry/d0.01_ef_eflag0.01_er0.01_no_repl_var_cov_0.2-1\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "path = os.path.abspath(os.getcwd()) + \"/\"+ report_directory\n",
    "print(path)\n",
    "big_df.to_csv(os.path.join(path, r\"all_dist_cov.csv\"), index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "JC correction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "import math\n",
    "def jc(b, do_correction, a):\n",
    "    deps = 0.0001\n",
    "    if abs(b) < deps:\n",
    "        return 0\n",
    "    loc = 1 - (4.0 * b / 3)\n",
    "\n",
    "    if (loc <= 0):\n",
    "        return 5.0 # max dist\n",
    "    if do_correction:\n",
    "        if a < deps:\n",
    "            a = deps\n",
    "        ret = a * (0.75 * (loc**(-1.0/a) - 1))\n",
    "    else:\n",
    "        ret = -0.75 * (math.log(loc))\n",
    "    return ret"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "do_correction = False; # \n",
    "alpha = 1.0;"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df_clean = pd.read_csv('primates_boot_0.8prct_withrepl/dimtrx_strap_0.txt', header = 0, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "col_names = df_clean.columns.tolist()[1:]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df_clean.head(n=10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#selects last column\n",
    "#df_clean.iloc[:, [-1]] \n",
    "\n",
    "#select all but first columns\n",
    "df_clean.iloc[:, 1:]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#df_clean[\"JC_correction\"] = df_clean[['value']].applymap(lambda x: jc(x, do_correction, alpha))\n",
    "\n",
    "#apply JC correction to entire dataframe except first column\n",
    "df_clean.iloc[:, 1:] = df_clean.iloc[:, 1:].applymap(lambda x: jc(x, do_correction, alpha))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df_clean.head(n=10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df_clean.to_csv(sep=' ', index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#invokes bash command\n",
    "%%bash\n",
    "fastme -i  dimat_testseq2.phy -o dimat_testseq2.tre"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#!fastme -i  temp.phy -o dimat_testseq2.tre\n",
    "#!fastme -i  dimat_simPrimates2.phy -o dimat_simPrimates2.tre"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#df_clean.to_string().split(\"\\n\")[1:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%bash \n",
    "# a=$report_directory\n",
    "# b='_tre'\n",
    "# dir=\"${a}${b}\"\n",
    "# mkdir \"${dir}\"\n",
    "# dir = tre\n",
    "# mkdir \"${dir}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "import treeswift"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_12.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_14.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_27.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_31.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_33.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_43.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_54.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_60.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_62.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_63.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_69.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_78.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_86.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "mtrx/dimtrx_strap_90.txt\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n",
      "\n",
      " . This analysis will run on 4 threads.\n",
      "\n",
      "#  Analysing dataset 1\n",
      "\n",
      " . Computing tree...\n",
      "\n",
      " . Time used 0h00m00s\n"
     ]
    }
   ],
   "source": [
    "support = 0\n",
    "\n",
    "# iterate over matrices\n",
    "for f in file_lst[:]:\n",
    "    #print(f)\n",
    "   \n",
    "    df_clean = pd.read_csv(f, header = 0, sep='\\t')\n",
    "    df_clean.iloc[:, 1:] = df_clean.iloc[:, 1:].applymap(lambda x: jc(x, do_correction, alpha))\n",
    "\n",
    "    ftmp = open(\"temp.phy\", \"w\")\n",
    "    ftmp.write (str(len(df_clean)))\n",
    "    ftmp.write ('\\n')\n",
    "    ftmp.close()\n",
    "    \n",
    "    #ftmp.write((df_clean.to_string().split(\"\\n\")[1:]).to_string())\n",
    "    #ftmp.write(df_clean.to_string())\n",
    "    \n",
    "    df_clean.to_csv(r'temp.phy', header=None, index=None, sep='\\t', mode = 'a')\n",
    "    \n",
    "    # compute trees\n",
    "    tre_name = f.split(\"/\")[-1] + \".tre\"\n",
    "    !fastme -i temp.phy -o tre/\"$tre_name\"\n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    tree = treeswift.read_tree_newick(\"tre/\"+tre_name)\n",
    "    \n",
    "    \n",
    "    hum = \"\"\n",
    "    chim = \"\"\n",
    "    gor = \"\"\n",
    "    orang = \"\"\n",
    "    \n",
    "    for label in tree.labels():               # include all nodes\n",
    "        #print(label)\n",
    "        if \"Human\" in label:\n",
    "            hum = label\n",
    "        if \"Chimp\" in label:\n",
    "            chim = label\n",
    "        if \"Gorilla\" in label:\n",
    "            gor = label\n",
    "        if \"Orang\" in label:\n",
    "            orang = label\n",
    "        \n",
    "    \n",
    "    i = 0\n",
    "    for node in tree.traverse_postorder():\n",
    "        #print(node)\n",
    "        if node.label == None:\n",
    "            node.label = str(i)\n",
    "        if node.is_root():\n",
    "            node.label = \"root\"\n",
    "            \n",
    "     \n",
    "    leaves_of_interest = {hum, chim}\n",
    "    m1 = tree.mrca(leaves_of_interest)\n",
    "    leaves_of_interest = {gor,orang}\n",
    "    m2 = tree.mrca(leaves_of_interest)\n",
    "    if m1 != m2:\n",
    "        support+=1\n",
    "    else:\n",
    "        print(f)\n",
    "    \n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "86\n"
     ]
    }
   ],
   "source": [
    "print(support)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [],
   "source": [
    "import dendropy\n",
    "tree1 = dendropy.Tree.get(path='tre/dimtrx_strap_81.txt.tre', schema=\"newick\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(simGorilla.fa_81:0.015164603722,(simHuman.fa_81:0.012861651293,simChimp.fa_81:0.012882550371):0.0023238492,simOrang.fa_81:0.03462685242)\n"
     ]
    }
   ],
   "source": [
    "print(tree1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "metadata": {},
   "outputs": [],
   "source": [
    "import treeswift\n",
    "tree = treeswift.read_tree_newick('tre/dimtrx_strap_81.txt.tre')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [],
   "source": [
    "map_all = tree.label_to_node(selection='all')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'': <treeswift.Node.Node object at 0x1141842d0>, 'simOrang.fa_81': <treeswift.Node.Node object at 0x114184990>, 'simChimp.fa_81': <treeswift.Node.Node object at 0x114184910>, 'simHuman.fa_81': <treeswift.Node.Node object at 0x1141847d0>, 'simGorilla.fa_81': <treeswift.Node.Node object at 0x11417eed0>}\n"
     ]
    }
   ],
   "source": [
    "print(map_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {},
   "outputs": [],
   "source": [
    "leaves_of_interest = {'simHuman.fa_81','simChimp.fa_81'}\n",
    "m = tree.mrca(leaves_of_interest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\n"
     ]
    }
   ],
   "source": [
    "print(m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "metadata": {},
   "outputs": [],
   "source": [
    "leaves_of_interest = {'simGorilla.fa_81','simOrang.fa_81'}\n",
    "m = tree.mrca(leaves_of_interest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "root\n"
     ]
    }
   ],
   "source": [
    "print(m.label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "simGorilla.fa_81\n",
      "simHuman.fa_81\n",
      "simChimp.fa_81\n",
      "3\n",
      "simOrang.fa_81\n",
      "root\n"
     ]
    }
   ],
   "source": [
    "i = 0\n",
    "for node in tree.traverse_postorder():\n",
    "    #print(node)\n",
    "    if node.label == None:\n",
    "        node.label = str(i)\n",
    "    if node.is_root():\n",
    "        node.label = \"root\"\n",
    "        \n",
    "    i+=1\n",
    "    \n",
    "    print(node.label)\n",
    "    \n",
    "    \n",
    "    \n",
    "    #print(node.get_parent())\n",
    "    #m = tree.mrca(map_all[node])\n",
    "    #print(map_all[node.get_parent])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = tree.mrca(\"simHuman.fa_81\", )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "simOrang.fa_81\n",
      "simChimp.fa_81\n",
      "simHuman.fa_81\n",
      "simGorilla.fa_81\n"
     ]
    }
   ],
   "source": [
    "for label in tree.labels():               # include all nodes\n",
    "    print(label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import treeswift\n",
    "tree = treeswift.read_tree_newick(my_newick_string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clean = pd.read_csv('dimat_flies.txt', header = 0, sep='\\t')\n",
    "df_clean = pd.melt(df_clean, id_vars=['sample'], value_vars= list(df_clean.columns[1:]) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clean[\"JC_correction\"] = df_clean[['value']].applymap(lambda x: jc(x, do_correction, alpha))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "my_newick_string = \"((simCow:0.18908,simDog:0.16303)sCow-sDog:0.032898,(simHuman:0.144018,(simMouse:0.084509,simRat:0.091589)sMouse-sRat:0.271974)sH-sM-sR:0.020593);\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " my_newick_string = \"((simGorilla:0.008825,(simHuman:0.0067,simChimp:0.006667)sHuman-sChimp:0.00225)sG-sH-sC:0.00968,simOrang:0.018318);\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "metadata": {},
   "outputs": [],
   "source": [
    "my_newick_string = \"((simCow:0.18908,simDog:0.16303)sCow-sDog:0.032898,(simHuman:0.144018,(simMouse:0.084509,simRat:0.091589)sMouse-sRat:0.271974)sH-sM-sR:0.020593);\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "read fly matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "metadata": {},
   "outputs": [],
   "source": [
    "import dendropy\n",
    "tree1 = dendropy.Tree.get(path='flies.tre', schema=\"newick\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "((droGri2:0.183954,droVir3:0.093575):0.0,(droMoj3:0.110563,((((droBip:0.034265,droAna3:0.042476):0.121927,(droKik:0.097564,((droFic:0.109823,(((dm3:0.023047,(droSim1:0.015485,droSec1:0.015184):0.01385):0.016088,(droYak2:0.026909,droEre2:0.029818):0.008929):0.047596,(droEug:0.102473,(droBia:0.069103,droTak:0.060723):0.015855):0.005098):0.010453):0.008044,(droEle:0.062413,droRho:0.051516):0.015405):0.046129):0.018695):0.078585,(droPer1:0.007065,dp4:0.0059):0.185269):0.068212,droWil1:0.259408):0.097093):0.03525)\n"
     ]
    }
   ],
   "source": [
    "print(tree1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "metadata": {},
   "outputs": [],
   "source": [
    "pdm = dendropy.PhylogeneticDistanceMatrix.from_tree(tree1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "metadata": {},
   "outputs": [],
   "source": [
    "my_dist_dict = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "metadata": {},
   "outputs": [],
   "source": [
    "column_names = [\"sample\", \"variable\", 'true_dist']\n",
    "\n",
    "df = pd.DataFrame(columns = column_names)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "metadata": {},
   "outputs": [],
   "source": [
    "my_dict = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.277529\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.32976700000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.619286\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.627497\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.579353\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.645785\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.633146\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.6394340000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.639133\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.629849\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.6327579999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.653986\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.636471\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.628091\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.605736\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.594839\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.576843\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.5756779999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.575705\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.277529\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.239388\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.528907\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.537118\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.488974\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.555406\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.542767\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.549055\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.548754\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.53947\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.542379\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.563607\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.546092\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.537712\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.515357\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.50446\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.48646399999999995\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.485299\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.48532600000000004\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.32976700000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.239388\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.510645\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.518856\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.470712\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.537144\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.524505\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.530793\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.530492\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.521208\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.5241169999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.545345\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.52783\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.51945\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.49709499999999995\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.486198\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.4682019999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.4670369999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.46706400000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.619286\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.528907\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.510645\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.076741\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.272451\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.33888300000000005\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.32624400000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.33253200000000005\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.332231\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.322947\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.32585600000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.34708400000000006\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.329569\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.32118900000000006\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.29883400000000004\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.287937\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.427111\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.425946\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.562397\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.627497\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.537118\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.518856\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.076741\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.280662\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.347094\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.33445500000000006\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.340743\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.340442\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.331158\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.334067\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.355295\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.33778\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.3294\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.307045\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.296148\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.435322\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.43415699999999996\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.570608\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.579353\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.488974\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.470712\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.272451\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.280662\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.26156\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.248921\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.255209\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.254908\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.24562399999999998\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.248533\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.26976100000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.25224599999999997\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.243866\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.22151099999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.210614\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.387178\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.386013\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.522464\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.645785\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.555406\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.537144\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.33888300000000005\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.347094\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.26156\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.207007\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.213295\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.212994\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.20371\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.206619\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.227847\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.21033200000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.20195200000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.195685\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.184788\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.45360999999999996\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.452445\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.5888960000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.633146\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.542767\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.524505\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.32624400000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.33445500000000006\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.248921\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.207007\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.052382000000000005\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.052081\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.07497300000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.077882\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.19430199999999997\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.176787\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.168407\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.18304600000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.172149\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.440971\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.43980600000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.576257\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.6394340000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.549055\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.530793\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.33253200000000005\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.340743\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.255209\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.213295\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.052382000000000005\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.030669000000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.081261\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.08417\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.20059\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.183075\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.174695\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.189334\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.178437\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.44725899999999996\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.446094\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.5825450000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.639133\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.548754\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.530492\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.332231\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.340442\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.254908\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.212994\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.052081\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.030669000000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.08096\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.083869\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.200289\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.182774\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.174394\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.189033\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.178136\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.44695799999999997\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.445793\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.582244\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.629849\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.53947\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.521208\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.322947\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.331158\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.24562399999999998\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.20371\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.07497300000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.081261\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.08096\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.056727\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.19100499999999998\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.17348999999999998\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.16510999999999998\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.179749\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.168852\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.437674\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.43650900000000004\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.57296\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.6327579999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.542379\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.5241169999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.32585600000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.334067\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.248533\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.206619\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.077882\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.08417\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.083869\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.056727\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.19391399999999998\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.176399\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.168019\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.18265800000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.171761\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.44058299999999995\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.439418\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.575869\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.653986\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.563607\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.545345\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.34708400000000006\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.355295\n",
      "<class 'str'>\n",
      "<class 'str'>\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.26976100000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.227847\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.19430199999999997\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.20059\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.200289\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.19100499999999998\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.19391399999999998\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.18743100000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.17905100000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.203886\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.19298900000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.46181099999999997\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.460646\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.597097\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.636471\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.546092\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.52783\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.329569\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.33778\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.25224599999999997\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.21033200000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.176787\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.183075\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.182774\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.17348999999999998\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.176399\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.18743100000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.129826\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.186371\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.17547400000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.444296\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.44313100000000005\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.579582\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.628091\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.537712\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.51945\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.32118900000000006\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.3294\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.243866\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.20195200000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.168407\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.174695\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.174394\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.16510999999999998\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.168019\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.17905100000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.129826\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.177991\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.16709400000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.43591599999999997\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.434751\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.571202\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.605736\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.515357\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.49709499999999995\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.29883400000000004\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.307045\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.22151099999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.195685\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.18304600000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.189334\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.189033\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.179749\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.18265800000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.203886\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.186371\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.177991\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.113929\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.41356099999999996\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.412396\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.5488470000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.594839\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.50446\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.486198\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.287937\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.296148\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.210614\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.184788\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.172149\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.178437\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.178136\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.168852\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.171761\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.19298900000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.17547400000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.16709400000000002\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.113929\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.402664\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.401499\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.53795\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.576843\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.48646399999999995\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.4682019999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.427111\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.435322\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.387178\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.45360999999999996\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.440971\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.44725899999999996\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.44695799999999997\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.437674\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.44058299999999995\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.46181099999999997\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.444296\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.43591599999999997\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.41356099999999996\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.402664\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.012965\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.519954\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.5756779999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.485299\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.4670369999999999\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.425946\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.43415699999999996\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.386013\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.452445\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.43980600000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.446094\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.445793\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.43650900000000004\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.439418\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.460646\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.44313100000000005\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.434751\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.412396\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.401499\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.012965\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.518789\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.575705\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.48532600000000004\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.46706400000000003\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.562397\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.570608\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.522464\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.5888960000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.576257\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.5825450000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.582244\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.57296\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.575869\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.597097\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.579582\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.571202\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.5488470000000001\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.53795\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.519954\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.518789\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "0.0\n",
      "<class 'str'>\n",
      "<class 'str'>\n"
     ]
    }
   ],
   "source": [
    "for idx1, taxon1 in enumerate(tree1.taxon_namespace):\n",
    "    for taxon2 in tree1.taxon_namespace:\n",
    "        a1 = str(taxon1.label)\n",
    "        a2 = str(taxon2.label)\n",
    "        d = pdm.patristic_distance(taxon1, taxon2)\n",
    "        print(d)\n",
    "        print(type(a1))\n",
    "        print(type(a2))\n",
    "        my_dict['sample'] = a1\n",
    "        my_dict['variable'] = a2\n",
    "        my_dict['true_dist'] = d\n",
    "        df = df.append(my_dict, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>true_dist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>droGri2</td>\n",
       "      <td>droGri2</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>droGri2</td>\n",
       "      <td>droVir3</td>\n",
       "      <td>0.277529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>droGri2</td>\n",
       "      <td>droMoj3</td>\n",
       "      <td>0.329767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>droGri2</td>\n",
       "      <td>droBip</td>\n",
       "      <td>0.619286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>droGri2</td>\n",
       "      <td>droAna3</td>\n",
       "      <td>0.627497</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample variable  true_dist\n",
       "0  droGri2  droGri2   0.000000\n",
       "1  droGri2  droVir3   0.277529\n",
       "2  droGri2  droMoj3   0.329767\n",
       "3  droGri2   droBip   0.619286\n",
       "4  droGri2  droAna3   0.627497"
      ]
     },
     "execution_count": 317,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(n=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "metadata": {},
   "outputs": [],
   "source": [
    "import treeswift\n",
    "tree = treeswift.read_tree_newick(my_newick_string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "simCow\n",
      "simDog\n",
      "sCow-sDog\n",
      "simHuman\n",
      "simMouse\n",
      "simRat\n",
      "sMouse-sRat\n",
      "sH-sM-sR\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for node in tree.traverse_postorder():\n",
    "    print(node)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [],
   "source": [
    "map_leaves = tree.label_to_node()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'': <treeswift.Node.Node object at 0x11d0ffb90>, 'simRat': <treeswift.Node.Node object at 0x11d104410>, 'simMouse': <treeswift.Node.Node object at 0x11d0aef90>, 'simHuman': <treeswift.Node.Node object at 0x11ce09ed0>, 'simDog': <treeswift.Node.Node object at 0x11d0ff290>, 'simCow': <treeswift.Node.Node object at 0x11d0ffe10>}\n"
     ]
    }
   ],
   "source": [
    "print(map_leaves)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [],
   "source": [
    "M = tree.distance_matrix()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{<treeswift.Node.Node object at 0x11d0ffe10>: {<treeswift.Node.Node object at 0x11d0ff290>: 0.35211000000000003, <treeswift.Node.Node object at 0x11ce09ed0>: 0.386589, <treeswift.Node.Node object at 0x11d0aef90>: 0.599054, <treeswift.Node.Node object at 0x11d104410>: 0.606134}, <treeswift.Node.Node object at 0x11d0ff290>: {<treeswift.Node.Node object at 0x11d0ffe10>: 0.35211000000000003, <treeswift.Node.Node object at 0x11ce09ed0>: 0.360539, <treeswift.Node.Node object at 0x11d0aef90>: 0.573004, <treeswift.Node.Node object at 0x11d104410>: 0.5800839999999999}, <treeswift.Node.Node object at 0x11d0aef90>: {<treeswift.Node.Node object at 0x11d104410>: 0.176098, <treeswift.Node.Node object at 0x11ce09ed0>: 0.500501, <treeswift.Node.Node object at 0x11d0ffe10>: 0.599054, <treeswift.Node.Node object at 0x11d0ff290>: 0.573004}, <treeswift.Node.Node object at 0x11d104410>: {<treeswift.Node.Node object at 0x11d0aef90>: 0.176098, <treeswift.Node.Node object at 0x11ce09ed0>: 0.507581, <treeswift.Node.Node object at 0x11d0ffe10>: 0.606134, <treeswift.Node.Node object at 0x11d0ff290>: 0.5800839999999999}, <treeswift.Node.Node object at 0x11ce09ed0>: {<treeswift.Node.Node object at 0x11d0aef90>: 0.500501, <treeswift.Node.Node object at 0x11d104410>: 0.507581, <treeswift.Node.Node object at 0x11d0ffe10>: 0.386589, <treeswift.Node.Node object at 0x11d0ff290>: 0.360539}}\n"
     ]
    }
   ],
   "source": [
    "print(M)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [],
   "source": [
    "#d = tree.distance_between(map_leaves['simCow'],map_leaves['simDog'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clean['true_dist'] = 0.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'dm3'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-239-d363bdbe6361>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m     \u001b[0md\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpdm\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpatristic_distance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sample'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variable'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m     \u001b[0mdf_clean\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mat\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'true_dist'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0md\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.7/site-packages/dendropy/calculate/phylogeneticdistance.py\u001b[0m in \u001b[0;36mpatristic_distance\u001b[0;34m(self, taxon1, taxon2, is_normalize_by_tree_size)\u001b[0m\n\u001b[1;32m    420\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mtaxon1\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mtaxon2\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    421\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0;36m0.0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 422\u001b[0;31m         \u001b[0md\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_taxon_phylogenetic_distances\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mtaxon1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mtaxon2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    423\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mis_normalize_by_tree_size\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    424\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0md\u001b[0m \u001b[0;34m/\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_tree_length\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'dm3'"
     ]
    }
   ],
   "source": [
    "for index, row in df_clean.iterrows():\n",
    "    df_clean.at[index, 'true_dist'] = tree.distance_between(map_leaves[row['sample']], map_leaves[row['variable']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "metadata": {},
   "outputs": [],
   "source": [
    "#flies\n",
    "new_df = pd.merge(df_clean, df,  how='left', left_on=['sample','variable'], right_on = ['sample','variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>value</th>\n",
       "      <th>JC_correction</th>\n",
       "      <th>true_dist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>dm3</td>\n",
       "      <td>dm3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>dp4</td>\n",
       "      <td>dm3</td>\n",
       "      <td>0.134899</td>\n",
       "      <td>0.148715</td>\n",
       "      <td>0.439806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>droAna3</td>\n",
       "      <td>dm3</td>\n",
       "      <td>0.123193</td>\n",
       "      <td>0.134576</td>\n",
       "      <td>0.334455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>droBia</td>\n",
       "      <td>dm3</td>\n",
       "      <td>0.098291</td>\n",
       "      <td>0.105356</td>\n",
       "      <td>0.176787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>droBip</td>\n",
       "      <td>dm3</td>\n",
       "      <td>0.121988</td>\n",
       "      <td>0.133136</td>\n",
       "      <td>0.326244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>droEle</td>\n",
       "      <td>dm3</td>\n",
       "      <td>0.100371</td>\n",
       "      <td>0.107754</td>\n",
       "      <td>0.183046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>droEre2</td>\n",
       "      <td>dm3</td>\n",
       "      <td>0.065768</td>\n",
       "      <td>0.068832</td>\n",
       "      <td>0.077882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>droEug</td>\n",
       "      <td>dm3</td>\n",
       "      <td>0.094712</td>\n",
       "      <td>0.101249</td>\n",
       "      <td>0.194302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>droFic</td>\n",
       "      <td>dm3</td>\n",
       "      <td>0.098526</td>\n",
       "      <td>0.105627</td>\n",
       "      <td>0.207007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>droGri2</td>\n",
       "      <td>dm3</td>\n",
       "      <td>0.163122</td>\n",
       "      <td>0.183942</td>\n",
       "      <td>0.633146</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample variable     value  JC_correction  true_dist\n",
       "0      dm3      dm3  0.000000       0.000000   0.000000\n",
       "1      dp4      dm3  0.134899       0.148715   0.439806\n",
       "2  droAna3      dm3  0.123193       0.134576   0.334455\n",
       "3   droBia      dm3  0.098291       0.105356   0.176787\n",
       "4   droBip      dm3  0.121988       0.133136   0.326244\n",
       "5   droEle      dm3  0.100371       0.107754   0.183046\n",
       "6  droEre2      dm3  0.065768       0.068832   0.077882\n",
       "7   droEug      dm3  0.094712       0.101249   0.194302\n",
       "8   droFic      dm3  0.098526       0.105627   0.207007\n",
       "9  droGri2      dm3  0.163122       0.183942   0.633146"
      ]
     },
     "execution_count": 319,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#flies\n",
    "new_df.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clean[\"err\"] = (df_clean[\"JC_correction\"]-df_clean[\"true_dist\"])/df_clean[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "#path = os.path.abspath(os.getcwd()) + \"/\"+ dir_name\n",
    "#print(path)\n",
    "#df.to_csv(os.path.join(path, r\"us_per_sample_results_k35_min32_v17.0_d4l2clmn7K15_tag2_t0.15.csv\"))\n",
    "\n",
    "#export_csv = df_clean.to_csv(r\"dimat_testseq.csv\", index = None, header=True)\n",
    "#export_csv = df_clean.to_csv(r\"dimat_simPrimates.csv\", index = None, header=True)\n",
    "export_csv = df_clean.to_csv(r\"dimat_simMammals.csv\", index = None, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "metadata": {},
   "outputs": [],
   "source": [
    "#flies\n",
    "new_df[\"err\"] = (new_df[\"JC_correction\"]-new_df[\"true_dist\"])/new_df[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 324,
   "metadata": {},
   "outputs": [],
   "source": [
    "#flies\n",
    "import os\n",
    "#path = os.path.abspath(os.getcwd()) + \"/\"+ dir_name\n",
    "#print(path)\n",
    "#df.to_csv(os.path.join(path, r\"us_per_sample_results_k35_min32_v17.0_d4l2clmn7K15_tag2_t0.15.csv\"))\n",
    "\n",
    "#export_csv = df_clean.to_csv(r\"dimat_testseq.csv\", index = None, header=True)\n",
    "#export_csv = df_clean.to_csv(r\"dimat_simPrimates.csv\", index = None, header=True)\n",
    "export_csv = new_df.to_csv(r\"dimat_flies.csv\", index = None, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_clean.rename(columns={'value':'dist_before_filt'}, inplace=True)\n",
    "df_clean.rename(columns={'value':'true_dist'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# flatten matrix for df after bbmerge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_af_dedup = pd.melt(df_af_dedup, id_vars=['sample'], value_vars= list(df_af_dedup.columns[1:]) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_af_dedup.rename(columns={'value':'dist_after_bbmerge'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'sub_Drosophila_bipectinata_2': 'Drosophila_bipectinata', 'sub_Drosophila_erecta_2': 'Drosophila_erecta', 'sub_Drosophila_ananassae_2': 'Drosophila_ananassae', 'sub_Drosophila_biarmipes_2': 'Drosophila_biarmipes', 'sub_Drosophila_mauritiana_2': 'Drosophila_mauritiana', 'sub_Drosophila_eugracilis_2': 'Drosophila_eugracilis', 'sub_Drosophila_mojavensis_2': 'Drosophila_mojavensis', 'sub_Drosophila_persimilis_2': 'Drosophila_persimilis', 'sub_Drosophila_simulans_2': 'Drosophila_simulans', 'sub_Drosophila_virilis_2': 'Drosophila_virilis', 'sub_Drosophila_pseudoobscura_2': 'Drosophila_pseudoobscura', 'sub_Drosophila_sechellia_2': 'Drosophila_sechellia', 'sub_Drosophila_willistoni_2': 'Drosophila_willistoni', 'sub_Drosophila_yakuba_2': 'Drosophila_yakuba'}\n"
     ]
    }
   ],
   "source": [
    "import csv\n",
    "filename =open(\"clean_sp_map.csv\", mode='r',encoding='utf8')\n",
    "fname_map = {}\n",
    "\n",
    "sp_map = pd.read_csv(filename, index_col=0, skiprows = 0).T.to_dict()\n",
    "\n",
    "for k, v in sp_map.items():\n",
    "    \n",
    "    #print(k.split(\".\")[0])\n",
    "    #new_k = k.split(\".\")[0]\n",
    "    #new_k = k + \"_1\"\n",
    "    new_k = \"sub_{}_2\".format(sp_map[k][\"sp\"])\n",
    "    \n",
    "    \n",
    "    fname_map[new_k] = sp_map[k][\"sp\"]\n",
    "    \n",
    "print(fname_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_af_dedup.replace({'sample': fname_map}, inplace = True)\n",
    "df_af_dedup.replace({'variable': fname_map}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.merge(df_clean, df_af_dedup,  how='left', left_on=['sample','variable'], right_on = ['sample','variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# flatten matrix for df after human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_af_human = pd.read_csv('ref_dist_mat_skmer_dros_0.1G_aft_human_wKraken.txt', header = 0, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_af_human = pd.melt(df_af_human, id_vars=['sample'], value_vars= list(df_af_human.columns[1:]) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_af_human.rename(columns={'value':'dist_after_human'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'ucseq_sub_Drosophila_bipectinata_2': 'Drosophila_bipectinata', 'ucseq_sub_Drosophila_erecta_2': 'Drosophila_erecta', 'ucseq_sub_Drosophila_ananassae_2': 'Drosophila_ananassae', 'ucseq_sub_Drosophila_biarmipes_2': 'Drosophila_biarmipes', 'ucseq_sub_Drosophila_mauritiana_2': 'Drosophila_mauritiana', 'ucseq_sub_Drosophila_eugracilis_2': 'Drosophila_eugracilis', 'ucseq_sub_Drosophila_mojavensis_2': 'Drosophila_mojavensis', 'ucseq_sub_Drosophila_persimilis_2': 'Drosophila_persimilis', 'ucseq_sub_Drosophila_simulans_2': 'Drosophila_simulans', 'ucseq_sub_Drosophila_virilis_2': 'Drosophila_virilis', 'ucseq_sub_Drosophila_pseudoobscura_2': 'Drosophila_pseudoobscura', 'ucseq_sub_Drosophila_sechellia_2': 'Drosophila_sechellia', 'ucseq_sub_Drosophila_willistoni_2': 'Drosophila_willistoni', 'ucseq_sub_Drosophila_yakuba_2': 'Drosophila_yakuba'}\n"
     ]
    }
   ],
   "source": [
    "lsh_map = {}\n",
    "for k, v in fname_map.items():\n",
    "    \n",
    "    new_key = \"ucseq_sub_{}_2\".format(v)\n",
    "    #print(new_key)\n",
    "    lsh_map[new_key] = fname_map[k]\n",
    "    \n",
    "print(lsh_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_af_human.replace({'sample': lsh_map}, inplace = True)\n",
    "df_af_human.replace({'variable': lsh_map}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.merge(new_df, df_af_human,  how='left', left_on=['sample','variable'], right_on = ['sample','variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# flatten matrices for df after lsh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.read_csv('dros_distances_updatedGTDB2.csv', header = 0, sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_lsh = pd.read_csv('ref_dist_mat_skmer_dros_0.1G_lsh_BacArchKraken.txt', header = 0, sep='\\t')\n",
    "#df_lsh = pd.read_csv('ref_dist_mat_skmer_dros0.1G_lsh_GTDB.txt', header = 0, sep='\\t')\n",
    "\n",
    "#df_lsh = pd.read_csv('distmtrx_dros_cnslt_GTDB_p4c2.txt', header = 0, sep='\\t')\n",
    "df_lsh = pd.read_csv('distmtrx_dros_cnslt_GTDB_p3c2.txt', header = 0, sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_lsh = pd.melt(df_lsh, id_vars=['sample'], value_vars= list(df_lsh.columns[1:]) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_lsh.rename(columns={'value':'dist_after_lsh_BacArchKraken'}, inplace=True)\n",
    "#df_lsh.rename(columns={'value':'dist_after_lsh_GTDB'}, inplace=True)\n",
    "#df_lsh.rename(columns={'value':'dist_after_lsh_GTDB_p4c2'}, inplace=True)\n",
    "df_lsh.rename(columns={'value':'dist_after_lsh_GTDB_p3c2'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'ucseq_ucseq_sub_Drosophila_bipectinata_2': 'Drosophila_bipectinata', 'ucseq_ucseq_sub_Drosophila_erecta_2': 'Drosophila_erecta', 'ucseq_ucseq_sub_Drosophila_ananassae_2': 'Drosophila_ananassae', 'ucseq_ucseq_sub_Drosophila_biarmipes_2': 'Drosophila_biarmipes', 'ucseq_ucseq_sub_Drosophila_mauritiana_2': 'Drosophila_mauritiana', 'ucseq_ucseq_sub_Drosophila_eugracilis_2': 'Drosophila_eugracilis', 'ucseq_ucseq_sub_Drosophila_mojavensis_2': 'Drosophila_mojavensis', 'ucseq_ucseq_sub_Drosophila_persimilis_2': 'Drosophila_persimilis', 'ucseq_ucseq_sub_Drosophila_simulans_2': 'Drosophila_simulans', 'ucseq_ucseq_sub_Drosophila_virilis_2': 'Drosophila_virilis', 'ucseq_ucseq_sub_Drosophila_pseudoobscura_2': 'Drosophila_pseudoobscura', 'ucseq_ucseq_sub_Drosophila_sechellia_2': 'Drosophila_sechellia', 'ucseq_ucseq_sub_Drosophila_willistoni_2': 'Drosophila_willistoni', 'ucseq_ucseq_sub_Drosophila_yakuba_2': 'Drosophila_yakuba'}\n"
     ]
    }
   ],
   "source": [
    "filt_map = {}\n",
    "for k, v in lsh_map.items():\n",
    "    \n",
    "    new_key = \"ucseq_{}\".format(k)\n",
    "    #print(new_key)\n",
    "    filt_map[new_key] = lsh_map[k]\n",
    "    \n",
    "print(filt_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_lsh.replace({'sample': filt_map}, inplace = True)\n",
    "df_lsh.replace({'variable': filt_map}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.merge(new_df, df_lsh,  how='left', left_on=['sample','variable'], right_on = ['sample','variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# flatten matrices for df after kraken"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken = pd.read_csv('ref_dist_mat_skmer_dros_0.1G_kraken_BacArchKraken.txt', header = 0, sep='\\t')\n",
    "#df_kraken = pd.read_csv('ref_dist_mat_skmer_dros_0.1G_kraken_GTDB.txt', header = 0, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken = pd.melt(df_kraken, id_vars=['sample'], value_vars= list(df_kraken.columns[1:]) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken.rename(columns={'value':'dist_after_kraken_BacArchKraken'}, inplace=True)\n",
    "#df_kraken.rename(columns={'value':'dist_after_kraken_GTDB'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken.replace({'sample': filt_map}, inplace = True)\n",
    "df_kraken.replace({'variable': filt_map}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>dist_after_kraken_BacArchKraken</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Drosophila_biarmipes</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.109156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Drosophila_bipectinata</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.064438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Drosophila_erecta</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.124482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Drosophila_eugracilis</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.112076</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample              variable  \\\n",
       "0    Drosophila_ananassae  Drosophila_ananassae   \n",
       "1    Drosophila_biarmipes  Drosophila_ananassae   \n",
       "2  Drosophila_bipectinata  Drosophila_ananassae   \n",
       "3       Drosophila_erecta  Drosophila_ananassae   \n",
       "4   Drosophila_eugracilis  Drosophila_ananassae   \n",
       "\n",
       "   dist_after_kraken_BacArchKraken  \n",
       "0                         0.000000  \n",
       "1                         0.109156  \n",
       "2                         0.064438  \n",
       "3                         0.124482  \n",
       "4                         0.112076  "
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_kraken.head(n=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.merge(new_df, df_kraken,  how='left', left_on=['sample','variable'], right_on = ['sample','variable'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "new_df.drop(columns=['err_after_human'], inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# compute relative error"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df[\"err_after_bbmerge\"] = (new_df[\"dist_after_bbmerge\"]-new_df[\"true_dist\"])/new_df[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df[\"err_after_human\"] = (new_df[\"dist_after_human\"]-new_df[\"true_dist\"])/new_df[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df[\"err_lsh_BacArchKraken\"] = (new_df[\"dist_after_lsh_BacArchKraken\"]-new_df[\"true_dist\"])/new_df[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df[\"err_kraken_BacArchKraken\"] = (new_df[\"dist_after_kraken_BacArchKraken\"]-new_df[\"true_dist\"])/new_df[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df[\"err_lsh_GTDB\"] = (new_df[\"dist_after_lsh_GTDB\"]-new_df[\"true_dist\"])/new_df[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df[\"err_lsh_GTDB_p4c2\"] = (new_df[\"dist_after_lsh_GTDB_p4c2\"]-new_df[\"true_dist\"])/new_df[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df[\"err_lsh_GTDB_p3c2\"] = (new_df[\"dist_after_lsh_GTDB_p3c2\"]-new_df[\"true_dist\"])/new_df[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df[\"err_kraken_GTDB\"] = (new_df[\"dist_after_kraken_GTDB\"]-new_df[\"true_dist\"])/new_df[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "#path = os.path.abspath(os.getcwd()) + \"/\"+ dir_name\n",
    "#print(path)\n",
    "#df.to_csv(os.path.join(path, r\"us_per_sample_results_k35_min32_v17.0_d4l2clmn7K15_tag2_t0.15.csv\"))\n",
    "\n",
    "export_csv = new_df.to_csv(r\"dros_distances.csv\", index = None, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "#add GTDB constructed with custom Taxonomy\n",
    "new_df = pd.read_csv('dros_distances.csv', header = 0, sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken = pd.read_csv('dimtrx_kraken_gtdbcusTax_Drosfilt_conf0.0.txt', header = 0, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken = pd.melt(df_kraken, id_vars=['sample'], value_vars= list(df_kraken.columns[1:]) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken.rename(columns={'value':'dist_after_kraken_GTDB_custTax'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken.replace({'sample': filt_map}, inplace = True)\n",
    "df_kraken.replace({'variable': filt_map}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>variable</th>\n",
       "      <th>dist_after_kraken_GTDB_custTax</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Drosophila_biarmipes</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.109738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Drosophila_bipectinata</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.064622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Drosophila_erecta</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.124940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Drosophila_eugracilis</td>\n",
       "      <td>Drosophila_ananassae</td>\n",
       "      <td>0.112595</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample              variable  \\\n",
       "0    Drosophila_ananassae  Drosophila_ananassae   \n",
       "1    Drosophila_biarmipes  Drosophila_ananassae   \n",
       "2  Drosophila_bipectinata  Drosophila_ananassae   \n",
       "3       Drosophila_erecta  Drosophila_ananassae   \n",
       "4   Drosophila_eugracilis  Drosophila_ananassae   \n",
       "\n",
       "   dist_after_kraken_GTDB_custTax  \n",
       "0                        0.000000  \n",
       "1                        0.109738  \n",
       "2                        0.064622  \n",
       "3                        0.124940  \n",
       "4                        0.112595  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_kraken.head(n=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.merge(new_df, df_kraken,  how='left', left_on=['sample','variable'], right_on = ['sample','variable'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df[\"err_kraken_GTDB_custTax\"] = (new_df[\"dist_after_kraken_GTDB_custTax\"]-new_df[\"true_dist\"])/new_df[\"true_dist\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "#path = os.path.abspath(os.getcwd()) + \"/\"+ dir_name\n",
    "#print(path)\n",
    "#df.to_csv(os.path.join(path, r\"us_per_sample_results_k35_min32_v17.0_d4l2clmn7K15_tag2_t0.15.csv\"))\n",
    "\n",
    "export_csv = new_df.to_csv(r\"dros_distances_updatedGTDB3.csv\", index = None, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "JC correction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "import math\n",
    "def jc(b, do_correction, a):\n",
    "    deps = 0.0001\n",
    "    if abs(b) < deps:\n",
    "        return 0\n",
    "    loc = 1 - (4.0 * b / 3)\n",
    "\n",
    "    if (loc <= 0):\n",
    "        return 5.0 # max dist\n",
    "    if do_correction:\n",
    "        if a < deps:\n",
    "            a = deps\n",
    "        ret = a * (0.75 * (loc**(-1.0/a) - 1))\n",
    "    else:\n",
    "        ret = -0.75 * (math.log(loc))\n",
    "    return ret"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "do_correction = False; # \n",
    "alpha = 1.0;"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "new_df2[\"JC_correction\"] = new_df2[['dist_after_kraken']].applymap(lambda x: jc(x, do_correction, alpha))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken = pd.read_csv('ref_dist_mat_skmer_crin_afkraken.txt', header = 0, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken.replace({'sample': lsh_map}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken = df_kraken.set_index('sample')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken.rename(columns=lsh_map, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove problematic taxon\n",
    "df_kraken.drop(columns=['C079_Promachocrinus_kerguelensis'], inplace=True)\n",
    "df_kraken.drop(['C079_Promachocrinus_kerguelensis'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken = df_kraken.applymap(lambda x: jc(x, do_correction, alpha))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_kraken.loc[:, df_kraken.columns != 'sample'].applymap(lambda x: jc(x, do_correction, alpha))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_kraken.to_csv(r\"crinoidea_jc2.csv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1314,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(dir_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_filenames = [\"d0.01/all_dist.csv\", \"d0.05/all_dist.csv\",\n",
    "                \"d0.10/all_dist.csv\", \"d0.15/all_dist.csv\", \"d0.25/all_dist.csv\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "metadata": {},
   "outputs": [],
   "source": [
    "#combine all files in the list\n",
    "combined_csv = pd.concat([pd.read_csv(f) for f in all_filenames ])\n",
    "#export to csv\n",
    "combined_csv.to_csv( \"combined_all_dist.csv\", index=False, encoding='utf-8-sig')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
